---
title: "Supplementary Analyses"
author: ["Shixiang Wang, Ziyu Tao, Tao Wu, Xue-Song Liu (Corresponding author)"]
date: "`r Sys.Date()`"
output:
  rmdformats::downcute:
    self_contained: true
    toc_depth: 3
bibliography: ref.bib
link-citations: yes
---


```{r setup-99, include=FALSE}
library(knitr)
library(rmdformats)
## Global options
options(max.print = "75")
opts_chunk$set(
  echo = FALSE,
  cache = FALSE,
  prompt = FALSE,
  tidy = "styler",
  comment = NA,
  dpi = 300,
  message = FALSE,
  warning = FALSE,
  echo = TRUE
)
opts_knit$set(width = 75)
Sys.setenv("LANGUAGE" = "EN")
```

In this part, we will add some supplementary analyses.

## CNS validation

### TCGA CNS

TCGA has ~10, 000 tumors and we can obtain allele specific copy number data from GDC portal, so it is the good resource to validate the CNS discovered in PCAWG.

Call signatures.

`PBS` file content:

```
#PBS -l walltime=1000:00:00
#PBS -l nodes=1:ppn=12
#PBS -S /bin/bash
#PBS -l mem=100gb
#PBS -j oe
#PBS -M w_shixiang@163.com
#PBS -q pub_fast

# Please set PBS arguments above
cd /public/home/wangshx/wangshx/PCAWG-TCGA
module load apps/R/3.6.1

echo Extracting TCGA signatures...

# Following are commands/scripts you want to run
# If you do not know how to set this, please check README.md file 
Rscript call_tcga.R
```

R script content:

```r
library(sigminer)

tally_X <- readRDS("tcga_cn_tally_X.rds")

sigprofiler_extract(tally_X$nmf_matrix, 
                    output = "TCGA_CN176X", 
                    range = 2:30, 
                    nrun = 100,
                    cores = 36,
                    init_method = "random",
                    is_exome = FALSE,
                    use_conda = TRUE)
```

Metadata:

```
THIS FILE CONTAINS THE METADATA ABOUT SYSTEM AND RUNTIME


-------System Info-------
Operating System Name: Linux
Nodename: node34
Release: 3.10.0-693.el7.x86_64
Version: #1 SMP Tue Aug 22 21:09:27 UTC 2017

-------Python and Package Versions------- 
Python Version: 3.8.5
Sigproextractor Version: 1.0.17
SigprofilerPlotting Version: 1.1.8
SigprofilerMatrixGenerator Version: 1.1.20
Pandas version: 1.1.2
Numpy version: 1.19.2
Scipy version: 1.5.2
Scikit-learn version: 0.23.2

--------------EXECUTION PARAMETERS--------------
INPUT DATA
	input_type: matrix
	output: TCGA_CN176X
	input_data: /tmp/RtmphYcBmk/dir852158ab0364/sigprofiler_input.txt
	reference_genome: GRCh37
	context_types: SBS176
	exome: False
NMF REPLICATES
	minimum_signatures: 2
	maximum_signatures: 30
	NMF_replicates: 100
NMF ENGINE
	NMF_init: random
	precision: single
	matrix_normalization: gmm
	resample: True
	seeds: random
	min_NMF_iterations: 10,000
	max_NMF_iterations: 1,000,000
	NMF_test_conv: 10,000
	NMF_tolerance: 1e-15
CLUSTERING
	clustering_distance: cosine
EXECUTION
	cpu: 36; Maximum number of CPU is 36
	gpu: False
Solution Estimation
	stability: 0.8
	min_stability: 0.2
	combined_stability: 1.0
COSMIC MATCH
	opportunity_genome: GRCh37
	nnls_add_penalty: 0.05
	nnls_remove_penalty: 0.01
	initial_remove_penalty: 0.05
	de_novo_fit_penalty: 0.02
	refit_denovo_signatures: False

-------Analysis Progress------- 
[2020-11-29 17:39:12] Analysis started: 

##################################

[2020-11-29 17:39:18] Analysis started for SBS176. Matrix size [176 rows x 10851 columns]

[2020-11-29 17:39:18] Normalization GMM with cutoff value set at 17600

[2020-11-29 19:12:03] SBS176 de novo extraction completed for a total of 2 signatures! 
Execution time:1:32:45

[2020-11-29 23:58:10] SBS176 de novo extraction completed for a total of 3 signatures! 
Execution time:4:46:06

[2020-11-30 08:32:21] SBS176 de novo extraction completed for a total of 4 signatures! 
Execution time:8:34:11

[2020-11-30 19:51:18] SBS176 de novo extraction completed for a total of 5 signatures! 
Execution time:11:18:56

[2020-12-01 07:10:18] SBS176 de novo extraction completed for a total of 6 signatures! 
Execution time:11:18:59

[2020-12-01 21:14:24] SBS176 de novo extraction completed for a total of 7 signatures! 
Execution time:14:04:06

[2020-12-02 15:48:43] SBS176 de novo extraction completed for a total of 8 signatures! 
Execution time:18:34:19

[2020-12-03 12:53:25] SBS176 de novo extraction completed for a total of 9 signatures! 
Execution time:21:04:41

[2020-12-04 11:25:25] SBS176 de novo extraction completed for a total of 10 signatures! 
Execution time:22:31:59

[2020-12-05 10:20:15] SBS176 de novo extraction completed for a total of 11 signatures! 
Execution time:22:54:50

[2020-12-06 11:36:03] SBS176 de novo extraction completed for a total of 12 signatures! 
Execution time:1 day, 1:15:48

[2020-12-07 14:43:31] SBS176 de novo extraction completed for a total of 13 signatures! 
Execution time:1 day, 3:07:28

[2020-12-09 01:15:22] SBS176 de novo extraction completed for a total of 14 signatures! 
Execution time:1 day, 10:31:50

[2020-12-10 08:02:39] SBS176 de novo extraction completed for a total of 15 signatures! 
Execution time:1 day, 6:47:17

[2020-12-11 16:05:31] SBS176 de novo extraction completed for a total of 16 signatures! 
Execution time:1 day, 8:02:51

[2020-12-13 02:19:40] SBS176 de novo extraction completed for a total of 17 signatures! 
Execution time:1 day, 10:14:09

[2020-12-14 14:56:13] SBS176 de novo extraction completed for a total of 18 signatures! 
Execution time:1 day, 12:36:32

[2020-12-16 02:39:41] SBS176 de novo extraction completed for a total of 19 signatures! 
Execution time:1 day, 11:43:27

[2020-12-18 02:54:09] SBS176 de novo extraction completed for a total of 20 signatures! 
Execution time:2 days, 0:14:28

[2020-12-19 20:20:57] SBS176 de novo extraction completed for a total of 21 signatures! 
Execution time:1 day, 17:26:47

[2020-12-21 19:30:07] SBS176 de novo extraction completed for a total of 22 signatures! 
Execution time:1 day, 23:09:10

[2020-12-24 00:44:41] SBS176 de novo extraction completed for a total of 23 signatures! 
Execution time:2 days, 5:14:34

[2020-12-25 19:03:07] SBS176 de novo extraction completed for a total of 24 signatures! 
Execution time:1 day, 18:18:25

[2020-12-27 07:50:06] SBS176 de novo extraction completed for a total of 25 signatures! 
Execution time:1 day, 12:46:58

[2020-12-29 04:40:22] SBS176 de novo extraction completed for a total of 26 signatures! 
Execution time:1 day, 20:50:16

[2020-12-30 12:35:04] SBS176 de novo extraction completed for a total of 27 signatures! 
Execution time:1 day, 7:54:41

[2020-12-31 19:04:04] SBS176 de novo extraction completed for a total of 28 signatures! 
Execution time:1 day, 6:29:00

[2021-01-02 04:50:07] SBS176 de novo extraction completed for a total of 29 signatures! 
Execution time:1 day, 9:46:02

[2021-01-03 14:15:32] SBS176 de novo extraction completed for a total of 30 signatures! 
Execution time:1 day, 9:25:25

[2021-01-03 14:48:21] Analysis ended: 

-------Job Status------- 
Analysis of mutational signatures completed successfully! 
Total execution time: 34 days, 21:09:09 
Results can be found in:  TCGA_CN176X  folder
```

Next we determine the signature number.

```{r eval=FALSE}
library(sigminer)
SP_TCGA <- sigprofiler_import("../SP/TCGA_CN176X/", order_by_expo = TRUE, type = "all")
saveRDS(SP_TCGA, file = "../data/TCGA/tcga_cn_solutions_sp.rds")
```


```{r}
SP_TCGA <- readRDS("../data/TCGA/tcga_cn_solutions_sp.rds")
```

```{r fig.width=9, fig.height=5}
show_sig_number_survey(
  SP_TCGA$all_stats %>%
    dplyr::rename(
      s = `Stability (Avg Silhouette)`,
      e = `Mean Cosine Distance`
    ) %>%
    dplyr::mutate(
      SignatureNumber = as.integer(gsub("[^0-9]", "", SignatureNumber))
    ),
  x = "SignatureNumber",
  left_y = "s", right_y = "e",
  left_name = "Stability (Avg Silhouette)",
  right_name = "Mean Cosine Distance",
  highlight = 20
)
```
We pick up 20 signatures for TCGA data due to its relatively high stability and low distance.


```{r eval=FALSE}
tcga_sigs <- SP_TCGA$solution_list$S20
saveRDS(tcga_sigs, file = "../data/TCGA/tcga_cn_sigs_CN176_signature.rds")
```

Let's go further check the similarities between PCAWG CNS and TCGA CNS.

```{r}
pcawg_sigs <- readRDS("../data/pcawg_cn_sigs_CN176_signature.rds")
tcga_sigs <- readRDS("../data/TCGA/tcga_cn_sigs_CN176_signature.rds")
```

Run similarity analysis.

```{r}
sim <- get_sig_similarity(pcawg_sigs, tcga_sigs)
```

```{r}
pheatmap::pheatmap(sim$similarity, cluster_rows = FALSE, cluster_cols = FALSE, display_numbers = TRUE)
```
From the pheatmap, we can see that:

- Most of PCAWG copy number signatures can be matched to TCGA copy number signatures with `>0.8` similarity.
- CNS4, CNS5, CNS6 and CNS14 from PCAWG has lower similarity to TCGA CNS. This indicates either they are not shared signatures or they have relatively different patterns in this two dataset.

It is not surprising to see that TCGA has more copy number signatures than PCAWG due to about 4 times more sample number.

This data remind us it is not easy to get perfect matched copy number signatures between different datasets due to the following reasons:

- TCGA copy number data are generated from SNP array and PCAWG copy number data are generated from WGS. 
- They also use different copy number calling softwares. Considering PCAWG data are consensus copy number data from many known calling software, the quality of PCAWG data is much higher than that of TCGA data.

Let's get the summary of match similarity.

```{r}
summary(apply(sim$similarity, 1, max))
```

### Merged CNS from clustering of cancer-type associated CNS

To further validate the CNS (stability) identified in PCAWG whole dataset. We also tried the cancer-type-wise signature identification pipeline from @degasperiPracticalFrameworkOnline2020, we implemented the procedure in our package `sigminer` by following the method description. [A benchmark](https://github.com/ShixiangWang/sigminer#key-interfaces-and-their-performances) has been done to make sure our implementation works properly.

1000 NMF runs (20 bootstrap catalogs and 50 NMF runs for each bootstrap catalog) was applied for data from each cancer type, which is similar to @degasperiPracticalFrameworkOnline2020. 

```{r eval=FALSE}
# This script is running on HPC
library(sigminer)

tally_X <- readRDS("pcawg_cn_tally_X.rds")
pcawg_types <- readRDS("pcawg_type_info.rds")

for (i in unique(pcawg_types$cancer_type)) {
  message("handling type: ", i)

  fn <- file.path("BP", paste0("PCAWG_CN176_", gsub("/", "-", i), ".rds"))

  if (!file.exists(fn)) {
    type_samples <- pcawg_types %>%
      dplyr::filter(cancer_type == i) %>%
      dplyr::pull(sample)
    message("Sample number: ", length(type_samples))

    sigs <- bp_extract_signatures(
      tally_X$nmf_matrix[type_samples, , drop = FALSE],
      range = 2:min(20, length(type_samples)-1),
      cores = 30,
      cores_solution = 20,
      cache_dir = "BP/BP_PCAWG_types",
      keep_cache = TRUE,
      only_core_stats = TRUE
    )
    saveRDS(sigs, file = fn)
  }
}
```

We then load the result and do manual inspection to determine the proper signature number for each cancer type.

```{r eval=FALSE}
library(sigminer)

cancer_type_files <- list.files("../data/cancer_types/", pattern = "PCAWG_CN176", full.names = TRUE)
pcawg <- lapply(cancer_type_files, readRDS)

bp_show_survey2(pcawg[[1]]) # 11
bp_show_survey2(pcawg[[2]]) # 14
bp_show_survey2(pcawg[[3]]) # 13
bp_show_survey2(pcawg[[4]]) # 8
bp_show_survey2(pcawg[[5]]) # 14
bp_show_survey2(pcawg[[6]]) # 11
bp_show_survey2(pcawg[[7]]) # 12
bp_show_survey2(pcawg[[8]]) # 8
bp_show_survey2(pcawg[[9]]) # 7
bp_show_survey2(pcawg[[10]]) # 5
bp_show_survey2(pcawg[[11]]) # 13
bp_show_survey2(pcawg[[12]]) # 9
bp_show_survey2(pcawg[[13]]) # 15
bp_show_survey2(pcawg[[14]]) # 7
bp_show_survey2(pcawg[[15]]) # 11
bp_show_survey2(pcawg[[16]]) # 12
bp_show_survey2(pcawg[[17]]) # 12
bp_show_survey2(pcawg[[18]]) # 17
bp_show_survey2(pcawg[[19]]) # 12
bp_show_survey2(pcawg[[20]]) # 6
bp_show_survey2(pcawg[[21]]) # 3
bp_show_survey2(pcawg[[22]]) # 7
bp_show_survey2(pcawg[[23]]) # 11
bp_show_survey2(pcawg[[24]]) # 19
bp_show_survey2(pcawg[[25]]) # 8
bp_show_survey2(pcawg[[26]]) # 10
bp_show_survey2(pcawg[[27]]) # 11
bp_show_survey2(pcawg[[28]]) # 9
bp_show_survey2(pcawg[[29]]) # 9
bp_show_survey2(pcawg[[30]]) # 15
bp_show_survey2(pcawg[[31]]) # 6
bp_show_survey2(pcawg[[32]]) # 17

type_sig <- data.frame(
  cancer_type = sub(".*_CN176_([^_]+).rds", "\\1", basename(cancer_type_files)),
  signum = c(11, 14, 13, 8, 14, 11, 12, 8,
             7, 5, 13, 9, 15, 7, 11, 12,
             12, 17, 12, 6, 3, 7, 11, 19,
             8, 10, 11, 9, 9, 15, 6, 17)
)

pcawg_sig_list <- lapply(
  seq_along(pcawg),
  function(i) {
    bp_get_sig_obj(pcawg[[i]], signum = type_sig$signum[i])
  }
)

names(pcawg_sig_list) <- type_sig$cancer_type
saveRDS(pcawg_sig_list, file = "data/pcawg_type_CNS.rds")
```

The signature number ranges from `3` to `19`.

Next we need to collapse all shared signatures from cancer types. To do this, we cluster all signatures based their cosine dissimilarity.

```{r}
library(sigminer)

pcawg_sig_list <- readRDS("../data/pcawg_type_CNS.rds")
pcawg_sig_list$`Biliary-AdenoCA`

cls <- bp_cluster_iter_list(pcawg_sig_list, k = 2:30)
```

We try `2` to `30` signatures here to check how the silhouette changes in different cluster number `k`.

```{r}
cls$sil_summary
plot(cls$sil_summary$k, cls$sil_summary$mean, type = "b", xlab = "Total signatures", ylab = "Mean silhouette")
```

In general, the mean silhouette is not high. But we can still see that `20` or `22` may be suitable. `k=5` is not considered because it is too small to explain our data.

Signature similarity is the key parameter in our study. Here we use an average cosine distance `0.4` (i.e. signatures with similarity `>=0.6` could be clustered as one) as a cutoff:

```{r fig.width=15}
factoextra::fviz_dend(cls$cluster, h = 0.4, cex = 0.15,
                      main = "",
                      xlab = "",
                      ylab = "1 - cosine similarity, average linkage")
```

Check the cluster number:

```{r}
cls_tree <- cutree(cls$cluster, h = 0.4)
table(cls_tree)
```

This is close to `22`. We observe there are a few isolated clusters with only one signature, they may be artifacts or should be assigned to other clusters but cannot due to low similarity to others.

Next we obtain clustered (grouped) signatures by averaging signatures in a cluster.

```{r}
grp_sigs <- bp_get_clustered_sigs(cls, cls_tree)
```

Next we load PCAWG global CNS and check their similarity.

```{r}
pcawg_sigs <- readRDS("../data/pcawg_cn_sigs_CN176_signature.rds")
```

```{r fig.width=8}
sim1 <- get_sig_similarity(pcawg_sigs, grp_sigs$grp_sigs)
sim_mat <- sim1$similarity
colnames(sim_mat) <- gsub("Sig", "Group-Sig", colnames(sim_mat))
pheatmap::pheatmap(sim_mat, cluster_rows = FALSE, cluster_cols = FALSE, display_numbers = TRUE)
```
We can see that most of PCAWG global signatures can be properly matched to group signatures.
`CNS2` å’Œ `CNS13` seems not stable and tend to be easier affected by extraction procedure.

Also, we observe almost all group signatures on the right side can be assigned to a PCAWG global signatures, although most of them have relatively low similarity. This data indicate the signatures may be redundant.

Let's check the similarity between group signatures and best matched global signatures.

```{r}
mc <- apply(sim1$similarity, 2, which.max)
mc_sim <- apply(sim1$similarity, 2, max)
mc_sim
```

Next let's visualize the data along with the similarity of signatures within a group (cluster).

```{r}
mc_df <- data.frame(
  grp_sig = names(mc),
  global_sig = paste0("CNS", mc),
  sim = as.numeric(mc_sim)
) %>%
  dplyr::mutate(
    label = paste0(grp_sig, "(", global_sig, ", ", sim, ")"),
    grp_sig = factor(grp_sig, levels = paste0("Sig", seq_len(nrow(.))))
  ) %>%
  dplyr::arrange(grp_sig)
```

```{r}
ggpubr::ggboxplot(grp_sigs$sim_sig_to_grp_mean,
                  x = "grp_sig", y = "sim", width = 0.3,
                  xlab = FALSE, ylab = "cosine similarity",
                  title = "Cosine similarity between signatures in each group") +
  ggpubr::rotate_x_text()
```

Add labels:

```{r}
ggpubr::ggboxplot(grp_sigs$sim_sig_to_grp_mean, #%>%
                  #dplyr::filter(grp_sig %in% select_sigs),
                  x = "grp_sig", y = "sim", width = 0.3,
                  xlab = FALSE, ylab = "Cosine similarity",
                  title = "Cosine similarity between signatures in each group") +
  ggpubr::rotate_x_text() +
  scale_x_discrete(labels = mc_df$label)
```

> Here y axis indicates cosine similarity between the averaging signature of each group and the individual signatures that belong to each group.

