<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <meta name="generator" content="pandoc" />

        <meta name="author" content="Shixiang Wang, Ziyu Tao, Tao Wu, Huimin Li, Xue-Song Liu (Corresponding author)" />
    
        <meta name="date" content="2020-12-31" />
    
    <title>Pan-Cancer Copy Number Signature Identification and Analysis</title>

        <script src="index_files/header-attrs-2.6/header-attrs.js"></script>
        <script src="index_files/jquery-1.12.4/jquery.min.js"></script>
        <meta name="viewport" content="width=device-width, initial-scale=1" />
        <link href="index_files/bootstrap-3.3.7/css/bootstrap.min.css" rel="stylesheet" />
        <script src="index_files/bootstrap-3.3.7/js/bootstrap.min.js"></script>
        <script src="index_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
        <script src="index_files/navigation-1.1/tabsets.js"></script>
        <link href="index_files/magnific-popup-1.1.0/magnific-popup.css" rel="stylesheet" />
        <script src="index_files/magnific-popup-1.1.0/jquery.magnific-popup.min.js"></script>
        <link href="index_files/downcute-0.1/downcute.css" rel="stylesheet" />
        <link href="index_files/downcute-0.1/downcute_fonts_embed.css" rel="stylesheet" />
        <script src="index_files/downcute-0.1/downcute.js"></script>
        <script src="index_files/prism-1.22/prism.js"></script>
        <script src="index_files/htmlwidgets-1.5.3/htmlwidgets.js"></script>
        <link href="index_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
        <script src="index_files/datatables-binding-0.16/datatables.js"></script>
        <link href="index_files/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
        <link href="index_files/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
        <script src="index_files/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
        <link href="index_files/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
        <script src="index_files/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
    
    
    
    
    <!-- tabsets -->
    <script>
      $(document).ready(function () {
	  window.buildTabsets("toc");
      });
      $(document).ready(function () {
	  $('.tabset-dropdown > .nav-tabs > li').click(function () {
	      $(this).parent().toggleClass('nav-tabs-open')
	  });
      });
    </script>

    <!-- code folding -->
    
    <!-- code download -->
    
    <!-- tabsets dropdown -->

    <style type="text/css">
      .tabset-dropdown > .nav-tabs {
	  display: inline-table;
	  max-height: 500px;
	  min-height: 44px;
	  overflow-y: auto;
	  background: white;
	  border: 1px solid #ddd;
	  border-radius: 4px;
      }
      
      .tabset-dropdown > .nav-tabs > li.active:before {
	  content: "";
	  font-family: 'Glyphicons Halflings';
	  display: inline-block;
	  padding: 10px;
	  border-right: 1px solid #ddd;
      }
      
      .tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
	  content: "&#xe258;";
	  border: none;
      }
      
      .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
	  content: "";
	  font-family: 'Glyphicons Halflings';
	  display: inline-block;
	  padding: 10px;
	  border-right: 1px solid #ddd;
      }
      
      .tabset-dropdown > .nav-tabs > li.active {
	  display: block;
      }

      .tabset-dropdown > .nav-tabs > li.active a {
  	  padding: 0 15px !important;
      }

      .tabset-dropdown > .nav-tabs > li > a,
      .tabset-dropdown > .nav-tabs > li > a:focus,
      .tabset-dropdown > .nav-tabs > li > a:hover {
	  border: none;
	  display: inline-block;
	  border-radius: 4px;
	  background-color: transparent;
      }
      
      .tabset-dropdown > .nav-tabs.nav-tabs-open > li {
	  display: block;
	  float: none;
      }
      
      .tabset-dropdown > .nav-tabs > li {
	  display: none;
	  margin-left: 0 !important;
      }
    </style>
    
</head>

<body class="preload">

   	
               <!-- downcute start -->   
   <div id="docute" class="Root">
     <div class="Page layout-narrow">
      <div class="Wrap">
        <div class="Sidebar">
          <div class="SidebarItems" id="toc">
            <ul>
            <li><a href="#license">LICENSE</a></li>
            <li><a href="#part-0-data-preprocessing">PART 0: Data preprocessing</a>
            <ul>
            <li><a href="#prepare-pcawg-datasets">Prepare PCAWG datasets</a></li>
            <li><a href="#generate-pcawg-sample-by-component-matrix">Generate PCAWG sample-by-component matrix</a></li>
            <li><a href="#prepare-tcga-datasets">Prepare TCGA datasets</a></li>
            <li><a href="#generate-tcga-sample-by-component-matrix">Generate TCGA sample-by-component matrix</a></li>
            <li><a href="#pcawg-genotypephenotype-data-type-and-source">PCAWG genotype/phenotype data type and source</a></li>
            </ul></li>
            <li><a href="#part-1-copy-number-signature-identification-and-activity-attribution">PART 1: Copy number signature identification and activity attribution</a>
            <ul>
            <li><a href="#copy-number-signature-identification">Copy number signature identification</a>
            <ul>
            <li><a href="#identification-pipeline">Identification pipeline</a></li>
            <li><a href="#apply-to-pcawg-catalog-matrix">Apply to PCAWG catalog matrix</a></li>
            <li><a href="#signature-number-determination">Signature number determination</a></li>
            <li><a href="#similarity-between-two-results-with-different-parameter-setting">Similarity between two results with different parameter setting</a></li>
            </ul></li>
            <li><a href="#reliable-signature-activity-attribution">Reliable signature activity attribution</a>
            <ul>
            <li><a href="#attribution-pipeline">Attribution pipeline</a></li>
            <li><a href="#apply-to-pcawg-catalog-matrix-and-signatures">Apply to PCAWG catalog matrix and signatures</a></li>
            </ul></li>
            <li><a href="#tcga-copy-number-signatures">TCGA copy number signatures</a></li>
            </ul></li>
            <li><a href="#part-2-signature-profile-visualization-and-analysis">PART 2: Signature profile visualization and analysis</a>
            <ul>
            <li><a href="#signature-profile-visualization">Signature profile visualization</a></li>
            <li><a href="#signature-analysis">Signature analysis</a>
            <ul>
            <li><a href="#signature-similarity">Signature similarity</a></li>
            <li><a href="#signature-activity-distribution">Signature activity distribution</a></li>
            </ul></li>
            <li><a href="#copy-number-profile-and-signature-activity">Copy number profile and signature activity</a>
            <ul>
            <li><a href="#copy-number-profile-for-representative-samples">Copy number profile for representative samples</a></li>
            <li><a href="#ideograph-for-copy-number-profile-reconstruction">Ideograph for copy number profile reconstruction</a></li>
            </ul></li>
            </ul></li>
            <li><a href="#part-3-pan-cancer-signature-analysis">PART 3: Pan-Cancer signature analysis</a>
            <ul>
            <li><a href="#signature-number-and-contribution-in-each-cancer-type">Signature number and contribution in each cancer type</a>
            <ul>
            <li><a href="#signature-activity-in-each-cancer-type">Signature activity in each cancer type</a></li>
            <li><a href="#signature-landscape">Signature landscape</a></li>
            </ul></li>
            <li><a href="#cancer-type-associated-enrichment">Cancer type associated enrichment</a></li>
            </ul></li>
            <li><a href="#part-4-cancer-subtyping-and-prognosis-analysis">PART 4: Cancer subtyping and prognosis analysis</a>
            <ul>
            <li><a href="#data-integration">Data integration</a></li>
            <li><a href="#clustering-with-signature-activity">Clustering with signature activity</a>
            <ul>
            <li><a href="#step-1-select-suitable-method-combination">Step 1: select suitable method combination</a></li>
            <li><a href="#step-2-select-suitable-cluster-number">Step 2: select suitable cluster number</a></li>
            </ul></li>
            <li><a href="#heatmap-for-clustering-and-genotypephenotype-features">Heatmap for clustering and genotype/phenotype features</a></li>
            <li><a href="#exploration-of-patients-prognosis-difference-across-clusters">Exploration of patients’ prognosis difference across clusters</a>
            <ul>
            <li><a href="#clean-data-for-survival-analysis">Clean data for survival analysis</a></li>
            <li><a href="#forest-plot">Forest plot</a></li>
            <li><a href="#k-m-plot">K-M plot</a></li>
            </ul></li>
            <li><a href="#the-difference-of-genotypephenotype-measures-across-clusters">The difference of genotype/phenotype measures across clusters</a>
            <ul>
            <li><a href="#signatures">Signatures</a></li>
            <li><a href="#other-integrated-variables">Other integrated variables</a></li>
            <li><a href="#driver-genes">Driver genes</a></li>
            </ul></li>
            </ul></li>
            <li><a href="#part-5-inference-of-copy-number-signature-etiologies">PART 5: Inference of copy number signature etiologies</a>
            <ul>
            <li><a href="#association-analysis-for-copy-number-signatures">Association analysis for copy number signatures</a>
            <ul>
            <li><a href="#copy-number-signature-activity-self-association">Copy number signature activity self-association</a></li>
            <li><a href="#copy-number-signatures-and-cosmic-signatures">Copy number signatures and COSMIC signatures</a></li>
            <li><a href="#signature-clustering">Signature clustering</a></li>
            <li><a href="#association-between-cns-and-genotypephenotype-features">Association between CNS and genotype/phenotype features</a></li>
            <li><a href="#association-between-cns-and-patients-prognosis">Association between CNS and patients’ prognosis</a></li>
            </ul></li>
            <li><a href="#signature-similarity-comparison-between-different-types-of-signatures">Signature similarity comparison between different types of signatures</a></li>
            <li><a href="#data-summary-and-inference-of-signature-etiologies">Data summary and inference of signature etiologies</a></li>
            </ul></li>
            <li><a href="#supplementary-analyses">Supplementary Analyses</a>
            <ul>
            <li><a href="#signature-association-network-construction-and-visualization">Signature association network construction and visualization</a></li>
            <li><a href="#tcga-cns-analysis">TCGA CNS analysis</a>
            <ul>
            <li><a href="#copy-number-signatures">Copy number signatures</a></li>
            <li><a href="#cancer-subtyping">Cancer subtyping</a></li>
            <li><a href="#prognosis">Prognosis</a></li>
            </ul></li>
            <li><a href="#exploration-of-differnt-signature-extraction-methods">Exploration of differnt signature extraction methods</a></li>
            </ul></li>
            <li><a href="#r-session">R Session:</a></li>
            <li><a href="#references">References</a></li>
            </ul>
          </div>
          <div data-position="sidebar:post-end" class="InjectedComponents"><div class="dark-theme-toggler"><div class="toggle"><div class="toggle-track"><div class="toggle-track-check"><img  src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAAlwSFlzAAALEwAACxMBAJqcGAAAAVlpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IlhNUCBDb3JlIDUuNC4wIj4KICAgPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4KICAgICAgPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIKICAgICAgICAgICAgeG1sbnM6dGlmZj0iaHR0cDovL25zLmFkb2JlLmNvbS90aWZmLzEuMC8iPgogICAgICAgICA8dGlmZjpPcmllbnRhdGlvbj4xPC90aWZmOk9yaWVudGF0aW9uPgogICAgICA8L3JkZjpEZXNjcmlwdGlvbj4KICAgPC9yZGY6UkRGPgo8L3g6eG1wbWV0YT4KTMInWQAABlJJREFUWAm1V3tsFEUcntnXvXu0tBWo1ZZHihBjCEWqkHiNaMLDRKOtQSKaiCFKQtS/SbxiFCHGCIkmkBSMwZhQNTFoQZD0DFiwtCDFAkdDqBBBKFj63rvdnfH7zfVo5aFBj0l2Z/dm5vd98/0es8dYjlpr62azufnDQNZcU1PciMfjWvb9rvZSMk4Ayfb36pLH13189GC8LAtIRLLPt+pzwrCuLq4ISEv/gHmitrAwfPbEkXc/ad4dL6iujrvyX0jcitgd/yZlZqftP6995Mr5TVLa22Tn8XVX2g/XLSRjUu7Q79jonS7I7hS7/0oOb5VyqF52n98oj7esXX07EjlxwXWisRmSnm3b29TTM8iYrjmFBWExubxwY/uhNas4r/WySl1fc5cetDMd7ydl+lMJJRw5WC8ud62Xx5rfepzwxgZmbhUYNS5Stvsj4yo2GXJEFBVHWDBkfdbR9HpYBaaUajDnBLKKpl1xRKYcgGtMCqEzTaSnThk/SQT0uJqTqFNBmXMCsZE48DzRZRMBRjv1GHNdk3HBImF9ZUvTyxM40pMKVc4JZBXQOLOFoDeKSxdp6HIQcO4rjYT9fn0pjbz9GLt7BAAODmjSVReXUMFzNW5x5vfxp2mIxZjIuQKJxAmFa+is2DQJJQ0JyBVExNOYcJnPxx/6/utnijmP555ALEagKAGGnGn64QORBjARcIA/yJk7JMJBLRrNtybTvH88KGjCf2jK86bhzmMcwDKFZEQvbIhxFYhChoMWMzU2iWznlIBEVJOsP+1bdX/ALx9l7jApADeDAEcMkE90JnUmmGl4USKQ0xhoW3JB5XY0YrxYWhLwMZZypUyjDGH35AbNwgUGiFBPpuGbHCpAOV1ZGXf2f/taftAv31DyeymN2d1IhAFAwTOmnzF/kKcdh3me7CYCOVNgycju84u8DeVlwfFq9/ZlTfldYrMUjOlrkjkD+rU+WzCROkcEchIDHR011syZW9JHD7y07N6JvhWMpz3pugaTkB6lWFVCKkhck0zzeMp2utq+uHrmfxOgoCO/Z8CXPlEQ1bdH8wgvhSIkEG0ICcQeExIFGdimjvKka7btJFZuaXOammIGKUCFQ53j9EN1dYKWqHf0t2w407W2tgs6h89ZnImjB55flh81tt9XirjjDuSl+oIPRQ0iWPgNZ5GqTqbBe3vSzEl5n5PhWKwocyR2HlqYN61qV18WjYjE8JLARZPQsUSim8foIRYTlGr02Ly7piASFRtKJ4VfieYhxdS2JcDVMN6xVOKZyrCGm8b108lrLRVzvptLH7IoEFLFANes6KnDi+uxfmvFnF17oALq5u1agu3/YfHkcSFzeSggV5eXRfIB7CHNcO5SUI+Ih5Ir7f4MAV9IqdFzdZgNpZw1Gcs1mNvgGbTbqQ9/cz7ZuuhgyYRQ49ljTyWHhr2DwpNHHFf+5gnWZ3Bharo+0TD5dNMw5vv9RlVpSRDHK4TlnoukhtYApuOHejSZQuo5g/A9BysdKRCyLl6062fN37OXMDlvUJtUrtmxo0avrW3wTrYs3jJ9RvRVChrmSmanPMpX2OXMsmDGh6AiEIwBAlvkOqIdBy+8JyAz8pz7QxiDth4KDy5uAlwzrWTnwC8Vc4KVAMZ3YUZ+IqoIjP3h5KFFX1ZMy3uW+7RhEDHgTi0zC9rS7uhPCDiNrGFyqBeERtKN/B0YlyFCkw0NJ5C0Ojv7zvT1a1WV1TuvZDdL4NTgB7CASYpsen6gqvG5jmTf5qHedADgkBl3D0nkSgNhZACDyi0FUKZRr3IdRjgN4WPPoFMIIegIK3mqd38fS80mcJKelM4szNyzZtQbkchGePuBRS8Eg9pHU8ojRQpSqs+ajAIwTjjUMQ/nvTNM0kicwYxZIYMh/891DYi+fvedB+c1xsm4lDU6ya+Axtz+RiAzEVYbajQOpq17F0R9QevNcEhfcU+xvyQQUalGJBSesqOkgPQ4YNyUZL9fSvUPDjoNAwN8/dwFjaczNkc3ptaMud1EIDtGcmXTcefO2cGSvKIFfp/2JIJxlq7xEl3nVPM4fDeIbPkD16/ptNc0bDu7qxbsu0R2JGywWMIjF2ft3tjfloAyQAGXiOn8hrqwbVvMXzaO+QeHXP6nF0wvX74Hf4NGG5GPjSlYoyM3P/0FbCT6zvM/yYoAAAAASUVORK5CYII=" role="presentation" style="pointer-events: none;" width="16" height="16"></div> <div class="toggle-track-x"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAAlwSFlzAAALEwAACxMBAJqcGAAAAVlpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IlhNUCBDb3JlIDUuNC4wIj4KICAgPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4KICAgICAgPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIKICAgICAgICAgICAgeG1sbnM6dGlmZj0iaHR0cDovL25zLmFkb2JlLmNvbS90aWZmLzEuMC8iPgogICAgICAgICA8dGlmZjpPcmllbnRhdGlvbj4xPC90aWZmOk9yaWVudGF0aW9uPgogICAgICA8L3JkZjpEZXNjcmlwdGlvbj4KICAgPC9yZGY6UkRGPgo8L3g6eG1wbWV0YT4KTMInWQAABwNJREFUWAmtV1tsFFUY/s6Z2d22zLYlZakUCRVaQcqlWIiCiS1gTEB9UAO+GR9En3iQGI0xJiSiRB98MjEq8cEQTSBeHhQM0V7whtEGDWC90BYitxahtNtu25058/v/ZzvLbilawJNM5+yZ89+//1LgJhYRNLW1uDfBAvpGiIk2O5auvfFxqIH3ZJ8/u06GN6Z9+wVl5SjcD1IbZa/UPkPyYl2uR4dreoD2bnbYxTlBBRytkHXtAREphP5KuH4lddx9h70yxX05t7yYXwGb6W8nx1jibpl2rFlGBxcG9M18okOrn7Bnk/BAO/4bI0UeEE1zjBp3UmvjOxJXJdaKN/ZiIu4tOZrAb4aTdZAZArKmWeiiJZ6jt5tiagdCS9+6cgO1Ne6Mvhe+ixTIfyDVhipnK9p+P0Edqx9RW/YZtQVGmOLChRxNNlyPsTEgPQKMB3dbEHa0h1awYmQ83enTd2vmUtvKd1Glv2RkzBb+kZGRrKtjzG60Wguhd/lJZBingbcfWWe72vjT75bJDrhYtvA0hrurETDr5HyF2Knb1MM4ab//xIoOqueA0edRnkkinTyJdYvqLFDZO4zUPFCvVoDjJq4T7TE61IWh4x5KqxX5KVKkX8WZ/t2ov2cb3MHt4dhIyOxIJxJOOF6xRx/99BksXLoecWcXytILMNBDqKpnGZWPquYfPxY8iXGR9fK+SgFrgcRPXPjVqhehL+3EmZ5RGJQi1QBU8TPThQnOQzm+5UXGIcetUeEAfP13VwzpI+w1jGJWdSliNfvVhiMPiOsllJag4M/UGHiqM6dlBb2OTLKHHV6KkvogrJ4XhBWniWK/Gp1MQyf93FOeUXKmKk/FzJxbQtKLjFXYT4USupy8fQVir2ynVEBiZMG0qtOHMS/AW4Gwrk7BG3C1F0B5nqNKE0CME4MfVRLPnXkBKe+ipvoFhNQywOhdghvLi0F8ReyVXV4BKTBRbbe5f64zR/DHsdZw1hJfeWlHl/GNRJzDxrd5m192z78TMaVnKELZoINZS4BzQ7vtnZljSnha/pPCbkuxzXcupYwI5tIeCpGc0Yp9tWHZQy/rmYhRfNgg4bHJBYLzGkxsRJF4XKlE2jBOHNSv3kY7Tj6vthzPFl61BrYwqFlmEQhtSVXmLiksxLmtRgYXI1ULU61JJ4eVKmG3/5sCVgpbMT6OMJ2E08/29Xf3w6v4FnHdCjfWgXu/O8Z5mLdCkeRs2khHe1DqOtQwbHWTAnM5S2HNmhALYo5KjkPFrMMKjZl6HxhWIAb0BqE+/73GrBRQUsKYiBu4JX8ycI6wtw+i5ef3NZpsrKVSHYCP37jwGDgeE1SA0S/xtl5SU2fs1ApEp0qTLVRjgyycDSsLHMSwmFltZMStR3uLLg6BdLhDa5dC6ryU2pHBe1BVO9tUcwfitJt2CLJZUHoG6T7Op75u0IyK31TCPcwFqgPk/KCaD3dFOuZBCO7xvCT/j048b3I3c7F2+WuOW7qdgkucFYlcQ4qop3yzTX7WaKfOCccye3Ts1Etq0+a/BHCF1yPgF3tAUkR6OrtGmo6gl94qqcXKh3rDyrOkPa58URoWcov2Mo6M+0QjrqKB+b7++oMa9Sz+ZkM0mie6aAtnGUvhmxaI+TogPOSQedgWioGSHFLn3v4kLh4HRspNmOGv41k+55siLFp2z6xYeJjhljFcbmxJlr4ga06TbevSByz/glQq4BJx46/c+237PbBqEYKxX3HpmKZEnQnr65X20hqJYaNcLoFOLiJk2LuBbyg7Q0OEn+hm0P3honxFD6rdxYorKpeIoi4YSSvyQHQIbM5t4+YNxLj/OxhVOOE4585qGpjnq+wSx6Q9CtNxTjd5klB+g6Mv36r0+b9cZFi44WYkHdG2ZWb3TtOUOXyVAlKlpGvJIAJ3eBMyfYS5C0qRZGtC85j+4sOasDe9xznPYezhhO/2Q6eP2fSOvYHOjtuQ1a9Q1VKynVDaMc8E0tptdxUsTFpFIYjcZKcbnoaQTNdiqCwNlL4G7oziSqGnT1ALf34vhk4R5zU3qYV9ONp9K88RtouShE68JwaU8dFw5W617shWa9ykeaBIn2hcsvPgL00k45QdTCZuSVcTRNs+8fnyLvooQfR5iujAnR9bxfY2xOVOxFS8SK3Le0l48VyYu1M8HRe5JD8wKPTjYnifaK3Wfn/GChYQ8ZAi6WRzWgqLV5YrsVLnZaVSoXU1g9gOIDwFySiGi+Zdrnzr7J3r+SMuszlcQCRn8lNGcTuSy2jOI7o9mxjZo+vR3ej3tN+ifRSOyUTS0+VMOid93cCubeiy/6TImS0QxRSCq2vxKr45zV+FQnjWH6D2xg+E9EatLcLAdHTgtGGD80D6jM0+aOl4wJgO/f96R2aJKCQ3yvgftRhdFMOpd6oAAAAASUVORK5CYII=" role="presentation" style="pointer-events: none;" width="16" height="16"></div></div> <div class="toggle-thumb"></div></div> <input type="checkbox" aria-label="Switch between Dark and Default theme" class="toggler-screen-reader-only"></div></div>
        </div>
        <div class="Main">
          <div class="Content" id="content"> 
   
   
        
      <h1 class="title">Pan-Cancer Copy Number Signature Identification and Analysis</h1>
      
      <p class="authors">
           <span class="glyphicon glyphicon-user"></span> Shixiang Wang, Ziyu Tao, Tao Wu, Huimin Li, Xue-Song Liu (Corresponding author)
      </p>
         <p class="date"><span class="glyphicon glyphicon-calendar"></span> 2020-12-31</p>
           

   
      
   
<!-- Don't indent these lines or it will mess pre blocks indentation --> 
<div class="page-content has-page-title">
<p><br></p>
<p>This document is compiled from an RMarkdown file which contains all code or description necessary to reproduce the analysis for the accompanying project. Each section below describes a different component of the analysis and most of numbers and figures are generated directly from the underlying data on compilation.</p>
<div id="license" class="section level1">
<h1>LICENSE</h1>
<p><strong>If you want to reuse the code in this report, please note the license below should be followed</strong>.</p>
<p>The code is made available for non commercial research purposes only under the MIT. However, notwithstanding any provision of the MIT License, the software currently may not be used for commercial purposes without explicit written permission after contacting Shixiang Wang <a href="mailto:wangshx@shanghaitech.edu.cn" class="email">wangshx@shanghaitech.edu.cn</a> or Xue-Song Liu <a href="mailto:liuxs@shanghaitech.edu.cn" class="email">liuxs@shanghaitech.edu.cn</a>.</p>
</div>
<div id="part-0-data-preprocessing" class="section level1">
<h1>PART 0: Data preprocessing</h1>
<p>In this part, raw data are collected from databases or papers and the core pre-processing steps are described in sections below.</p>
<p><em>The pre-processing work has been done by setting root path of the project repository as work directory. Therefore, keep in mind the work directory should be properly set if you are interested in reproducing the pre-processing procedure.</em></p>
<div id="prepare-pcawg-datasets" class="section level2">
<h2>Prepare PCAWG datasets</h2>
<p>We download PCAWG phenotype and copy number data from <a href="https://xenabrowser.net/datapages/">UCSC Xena</a> and save them to local machine with R format.</p>
<pre class="r"><code>dir.create(&quot;Xena&quot;)

# Phenotype (Specimen Centric) -----------------------------------------------

library(UCSCXenaTools)
pcawg_phenotype &lt;- XenaData %&gt;%
  dplyr::filter(XenaHostNames == &quot;pcawgHub&quot;) %&gt;%
  XenaScan(&quot;phenotype&quot;) %&gt;%
  XenaScan(&quot;specimen centric&quot;) %&gt;%
  XenaGenerate() %&gt;%
  XenaQuery()
pcawg_phenotype &lt;- pcawg_phenotype %&gt;%
  XenaDownload(destdir = &quot;Xena&quot;, trans_slash = TRUE)

phenotype_list &lt;- XenaPrepare(pcawg_phenotype)
pcawg_samp_info_sp &lt;- phenotype_list[1:4]

saveRDS(pcawg_samp_info_sp, file = &quot;data/pcawg_samp_info_sp.rds&quot;)

# PCAWG (Specimen Centric) ------------------------------------------------

download.file(
  &quot;https://pcawg.xenahubs.net/download/20170119_final_consensus_copynumber_sp.gz&quot;,
  &quot;Xena/pcawg_copynumber_sp.gz&quot;
)

pcawg_cn &lt;- data.table::fread(&quot;Xena/pcawg_copynumber_sp.gz&quot;)
pcawg_samp_info_sp &lt;- readRDS(&quot;data/pcawg_samp_info_sp.rds&quot;)

sex_dt &lt;- pcawg_samp_info_sp$pcawg_donor_clinical_August2016_v9_sp %&gt;%
  dplyr::select(xena_sample, donor_sex) %&gt;%
  purrr::set_names(c(&quot;sample&quot;, &quot;sex&quot;)) %&gt;%
  data.table::as.data.table()

saveRDS(sex_dt, file = &quot;data/pcawg_sex_sp.rds&quot;)

pcawg_cn &lt;- pcawg_cn[!is.na(total_cn)]
pcawg_cn$value &lt;- NULL
pcawg_cn &lt;- pcawg_cn[, c(1:5, 7)]
colnames(pcawg_cn)[1:5] &lt;- c(&quot;sample&quot;, &quot;Chromosome&quot;, &quot;Start.bp&quot;, &quot;End.bp&quot;, &quot;modal_cn&quot;)

saveRDS(pcawg_cn, file = &quot;data/pcawg_copynumber_sp.rds&quot;)

# LOH ---------------------------------------------------------------------

pcawg_cn &lt;- readRDS(&quot;data/pcawg_copynumber_sp.rds&quot;)

pcawg_loh &lt;- pcawg_cn %&gt;%
  dplyr::filter(Chromosome %in% as.character(1:22)) %&gt;%
  dplyr::mutate(len = End.bp - Start.bp + 1) %&gt;%
  dplyr::group_by(sample) %&gt;%
  dplyr::summarise(
    n_LOH = sum(minor_cn == 0 &amp; modal_cn &gt; 0 &amp; len &gt;= 1e4)
  ) %&gt;%
  setNames(c(&quot;sample&quot;, &quot;n_LOH&quot;)) %&gt;%
  data.table::as.data.table()

saveRDS(pcawg_loh, file = &quot;data/pcawg_loh.rds&quot;)</code></pre>
</div>
<div id="generate-pcawg-sample-by-component-matrix" class="section level2">
<h2>Generate PCAWG sample-by-component matrix</h2>
<p>We then read the copy number data and transform it into a <code>CopyNumber</code> object with R package sigminer. Then <code>sig_tally()</code> is used to generate the matrix for NMF decomposition.</p>
<pre class="r"><code>library(sigminer)

# Only focus autosomes, suggested by Prof. Liu
pcawg_cn &lt;- readRDS(&quot;data/pcawg_copynumber_sp.rds&quot;)
table(pcawg_cn$Chromosome)
pcawg_cn &lt;- pcawg_cn[!Chromosome %in% c(&quot;X&quot;, &quot;Y&quot;)]
cn_obj &lt;- read_copynumber(pcawg_cn,
  add_loh = TRUE,
  loh_min_len = 1e4,
  loh_min_frac = 0.05,
  max_copynumber = 1000L,
  genome_build = &quot;hg19&quot;,
  complement = FALSE,
  genome_measure = &quot;called&quot;
)
saveRDS(cn_obj, file = &quot;data/pcawg_cn_obj.rds&quot;)

# Tally -------------------------------------------------------------------
library(sigminer)
cn_obj &lt;- readRDS(&quot;data/pcawg_cn_obj.rds&quot;)
table(cn_obj@data$chromosome)

tally_X &lt;- sig_tally(cn_obj,
  method = &quot;X&quot;,
  add_loh = TRUE,
  cores = 10
)
saveRDS(tally_X, file = &quot;data/pcawg_cn_tally_X.rds&quot;)
str(tally_X$all_matrices, max.level = 1)

p &lt;- show_catalogue(tally_X,
  mode = &quot;copynumber&quot;, method = &quot;X&quot;,
  style = &quot;cosmic&quot;, y_tr = function(x) log10(x + 1),
  y_lab = &quot;log10(count +1)&quot;
)
p
ggplot2::ggsave(&quot;output/pcawg_catalogs_tally_X.pdf&quot;, plot = p, width = 16, height = 2.5)

head(sort(colSums(tally_X$nmf_matrix)), n = 50)

## classes without LOH labels
tally_X_noLOH &lt;- sig_tally(cn_obj,
  method = &quot;X&quot;,
  add_loh = FALSE,
  cores = 10
)
saveRDS(tally_X_noLOH$all_matrices, file = &quot;data/pcawg_cn_tally_X_noLOH.rds&quot;)
str(tally_X_noLOH$all_matrices, max.level = 1)

p &lt;- show_catalogue(tally_X_noLOH,
  mode = &quot;copynumber&quot;, method = &quot;X&quot;,
  style = &quot;cosmic&quot;, y_tr = function(x) log10(x + 1),
  y_lab = &quot;log10(count +1)&quot;
)
p
ggplot2::ggsave(&quot;output/pcawg_catalogs_tally_X_noLOH.pdf&quot;, plot = p, width = 16, height = 2.5)</code></pre>
<p>The generated catalog profiles for LOH version and non-LOH version component classification can be viewed by the following link:</p>
<ul>
<li><a href="https://github.com/ShixiangWang/CNX/blob/master/output/pcawg_catalogs_tally_X.pdf">pcawg_catalogs_tally_X.pdf</a> - 176 components</li>
<li><a href="https://github.com/ShixiangWang/CNX/blob/master/output/pcawg_catalogs_tally_X_noLOH.pdf">pcawg_catalogs_tally_X_noLOH.pdf</a> - 136 components</li>
</ul>
</div>
<div id="prepare-tcga-datasets" class="section level2">
<h2>Prepare TCGA datasets</h2>
<p>TCGA allele-specific copy number data are downloaded from <a href="https://portal.gdc.cancer.gov/">GDC portal</a> and transformed into R format by Huimin Li.</p>
<p>The data is go further checked and cleaned by Shixiang.</p>
<pre class="r"><code># Huimin collected TCGA data from GDC portal
# and generate file with format needed by sigminer
# Here I will firstly further clean up the data

library(data.table)
x &lt;- readRDS(&quot;data/TCGA/datamicopy.rds&quot;)
setDT(x)

colnames(x)[6] &lt;- &quot;minor_cn&quot;
head(x)
x[, sample := substr(sample, 1, 15)]
saveRDS(x, file = &quot;data/TCGA/tcga_cn.rds&quot;)

rm(list = ls())</code></pre>
<p>TCGA phenotype data is downloaded from UCSC Xena.</p>
<pre class="r"><code>download.file(
  &quot;https://pancanatlas.xenahubs.net/download/Survival_SupplementalTable_S1_20171025_xena_sp.gz&quot;,
  &quot;Xena/Survival_SupplementalTable_S1_20171025_xena_sp.gz&quot;
)

tcga_cli &lt;- data.table::fread(&quot;Xena/Survival_SupplementalTable_S1_20171025_xena_sp.gz&quot;)
table(tcga_cli$`cancer type abbreviation`)
saveRDS(tcga_cli, file = &quot;data/TCGA/tcga_cli.rds&quot;)</code></pre>
</div>
<div id="generate-tcga-sample-by-component-matrix" class="section level2">
<h2>Generate TCGA sample-by-component matrix</h2>
<p>Similar to PCAWG, TCGA matrices are also generated.</p>
<pre class="r"><code>library(sigminer)

# Only focus autosomes
tcga_cn &lt;- readRDS(&quot;data/TCGA/tcga_cn.rds&quot;)
table(tcga_cn$chromosome)
tcga_cn &lt;- tcga_cn[!chromosome %in% c(&quot;chrX&quot;, &quot;chrY&quot;)]
cn_obj &lt;- read_copynumber(
  tcga_cn,
  seg_cols = c(&quot;chromosome&quot;, &quot;start&quot;, &quot;end&quot;, &quot;segVal&quot;),
  add_loh = TRUE,
  loh_min_len = 1e4,
  loh_min_frac = 0.05,
  max_copynumber = 1000L,
  genome_build = &quot;hg38&quot;,
  complement = FALSE,
  genome_measure = &quot;called&quot;
)
saveRDS(cn_obj, file = &quot;data/TCGA/tcga_cn_obj.rds&quot;)

# Tally step
# Generate the counting matrices
# Same as what have done for PCAWG dataset
library(sigminer)
cn_obj &lt;- readRDS(&quot;data/tcga_cn_obj.rds&quot;)
table(cn_obj@data$chromosome)

tally_X &lt;- sig_tally(cn_obj,
  method = &quot;X&quot;,
  add_loh = TRUE,
  cores = 10
)
saveRDS(tally_X, file = &quot;data/TCGA/tcga_cn_tally_X.rds&quot;)


p &lt;- show_catalogue(tally_X,
  mode = &quot;copynumber&quot;, method = &quot;X&quot;,
  style = &quot;cosmic&quot;, y_tr = function(x) log10(x + 1),
  y_lab = &quot;log10(count +1)&quot;
)
p
ggplot2::ggsave(&quot;output/tcga_catalogs_tally_X.pdf&quot;, plot = p, width = 16, height = 2.5)

head(sort(colSums(tally_X$nmf_matrix)), n = 50)

## classes without LOH labels
tally_X_noLOH &lt;- sig_tally(cn_obj,
  method = &quot;X&quot;,
  add_loh = FALSE,
  cores = 10
)
saveRDS(tally_X_noLOH$all_matrices, file = &quot;data/TCGA/tcga_cn_tally_X_noLOH.rds&quot;)
str(tally_X_noLOH$all_matrices, max.level = 1)

p &lt;- show_catalogue(tally_X_noLOH,
  mode = &quot;copynumber&quot;, method = &quot;X&quot;,
  style = &quot;cosmic&quot;, y_tr = function(x) log10(x + 1),
  y_lab = &quot;log10(count +1)&quot;
)
p
ggplot2::ggsave(&quot;output/tcga_catalogs_tally_X_noLOH.pdf&quot;, plot = p, width = 16, height = 2.5)</code></pre>
<p>The generated catalog profiles for LOH version and non-LOH version component classification can be viewed by the following link:</p>
<ul>
<li><a href="https://github.com/ShixiangWang/CNX/blob/master/output/tcga_catalogs_tally_X.pdf">tcga_catalogs_tally_X.pdf</a> - 176 components</li>
<li><a href="https://github.com/ShixiangWang/CNX/blob/master/output/tcga_catalogs_tally_X_noLOH.pdf">tcga_catalogs_tally_X_noLOH.pdf</a> - 136 components</li>
</ul>
</div>
<div id="pcawg-genotypephenotype-data-type-and-source" class="section level2">
<h2>PCAWG genotype/phenotype data type and source</h2>
<p>If no references described, PCAWG genotype/phenotype data used in this study are obtained from UCSC Xena database, the original data source is <a href="https://www.nature.com/collections/afdejfafdb">PCAWG paper collection</a> published in early 2020.</p>
<ul>
<li>Consensus copy number (<a href="https://xenabrowser.net/datapages/?dataset=20170119_final_consensus_copynumber_sp&amp;host=https%3A%2F%2Fpcawg.xenahubs.net&amp;removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443">page in UCSC Xena</a>).</li>
<li>General phenotype information (<a href="https://xenabrowser.net/datapages/?cohort=PCAWG%20(specimen%20centric)&amp;removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443">page in UCSC Xena</a>).</li>
<li>Tumor purity &amp; ploidy &amp; WGD status (<a href="https://xenabrowser.net/datapages/?dataset=consensus.20170217.purity.ploidy_sp&amp;host=https%3A%2F%2Fpcawg.xenahubs.net&amp;removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443">page in UCSC Xena</a>).</li>
<li>Gene level fusion events (<a href="https://xenabrowser.net/datapages/?dataset=pcawg3_fusions_PKU_EBI.gene_centric.sp.xena&amp;host=https%3A%2F%2Fpcawg.xenahubs.net&amp;removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443">page in UCSC Xena</a>).</li>
<li>LOH number, calculated from copy number data, defined as number of copy number segments passing the following criterion:
<ul>
<li>Minor copy number equals to <code>0</code>.</li>
<li>Total copy number is greater than <code>0</code>.</li>
<li>Segment length is greater than <code>10Kb</code>.</li>
</ul></li>
<li>HRD status by CHORD framework, the raw data is available at <span class="citation"><a href="#ref-nguyenPancancerLandscapeHomologous2020" role="doc-biblioref">Nguyen et al.</a> (<a href="#ref-nguyenPancancerLandscapeHomologous2020" role="doc-biblioref">2020</a>)</span>, cleaned by Huimin Li.</li>
<li>Chromothripsis detected by ShatterSeek, the data is obtained from <a href="http://compbio.med.harvard.edu/chromothripsis/" class="uri">http://compbio.med.harvard.edu/chromothripsis/</a>.</li>
<li>Amplicons (including ecDNA) detected by AmpliconArchitect, obtained from <span class="citation"><a href="#ref-kimExtrachromosomalDNAAssociated2020" role="doc-biblioref">Kim et al.</a> (<a href="#ref-kimExtrachromosomalDNAAssociated2020" role="doc-biblioref">2020</a>)</span>.</li>
<li>Telomere content detected by TelomereHunter, obtained from <span class="citation"><a href="#ref-pcawg-structuralvariationworkinggroupGenomicFootprintsActivated2020" role="doc-biblioref">PCAWG-Structural Variation Working Group et al.</a> (<a href="#ref-pcawg-structuralvariationworkinggroupGenomicFootprintsActivated2020" role="doc-biblioref">2020</a>)</span>.</li>
<li>APOBEC mutations (<a href="https://xenabrowser.net/datapages/?dataset=MAF_Aug31_2016_sorted_A3A_A3B_comparePlus.sp&amp;host=https%3A%2F%2Fpcawg.xenahubs.net&amp;removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443">page in UCSC Xena</a>).</li>
</ul>
</div>
</div>
<div id="part-1-copy-number-signature-identification-and-activity-attribution" class="section level1">
<h1>PART 1: Copy number signature identification and activity attribution</h1>
<p>In this part, how the copy number signatures are identified and how the corresponding signature activities (a.k.a. exposures) are attributed are described in sections below. <strong>The 176 component classification for PCAWG data is our main focus.</strong></p>
<p><em>The work of this part has been done by either setting root path of the project repository as work directory or moving data and code to the same path (see below). Therefore, keep in mind the work directory/data path should be properly set if you are interested in reproducing the analysis procedure.</em></p>
<p><em>The signature identification work cannot be done in local machine because high intensity calculations are required, we used HPC of ShanghaiTech University to submit computation jobs to HPC clusters. Input data files, R scripts and PBS job scripts are stored in a same path, the structure can be viewed as:</em></p>
<pre class="sh"><code>.
├── 04-call-tcga-signatures-with-BP-types.R
├── 05-call-pcawg-signatures-with-BP-types.R
├── bp_pcawg.pbs
├── bp_pcawg_types.pbs
├── bp_tcga.pbs
├── bp_tcga_types.pbs
├── call_pcawg_bp.R
├── call_pcawg_sp.R
├── call_tcga_bp.R
├── call_tcga.R
├── pcawg_cn_tally_X.rds
├── pcawg.pbs
├── tcga_cn_tally_X.rds
├── tcga.pbs</code></pre>
<div id="copy-number-signature-identification" class="section level2">
<h2>Copy number signature identification</h2>
<p>The signature identification pipeline is adopted from <span class="citation"><a href="#ref-degasperiPracticalFrameworkOnline2020" role="doc-biblioref">Degasperi et al.</a> (<a href="#ref-degasperiPracticalFrameworkOnline2020" role="doc-biblioref">2020</a>)</span>, we implemented the procedure in our package sigminer by following the method description. <a href="https://github.com/ShixiangWang/sigminer#key-interfaces-and-their-performances">A benchmark</a> has been done to make sure our implementation works properly.</p>
<div id="identification-pipeline" class="section level3">
<h3>Identification pipeline</h3>
<p>To make the signature identification procedure more clear, we drew a flowchart. We name this pipeline “Best Practice” and use its short name <em>BP</em> (or <em>bp</em>).</p>
<blockquote>
<p>TODO: add flowchart</p>
</blockquote>
</div>
<div id="apply-to-pcawg-catalog-matrix" class="section level3">
<h3>Apply to PCAWG catalog matrix</h3>
<p>PBS file <code>bp_pcawg.pbs</code> content:</p>
<pre class="sh"><code>#PBS -l walltime=500:00:00
#PBS -l nodes=1:ppn=50
#PBS -S /bin/bash
#PBS -l mem=300gb
#PBS -j oe
#PBS -M w_shixiang@163.com
#PBS -q slst_pub

# Please set PBS arguments above
cd /public/home/wangshx/wangshx/PCAWG-TCGA
module load apps/R/3.6.1

# Following are commands/scripts you want to run
Rscript call_pcawg_bp.R</code></pre>
<p>R script <code>call_pcawg_bp.R</code> content:</p>
<pre class="r"><code>library(sigminer)

tally_X &lt;- readRDS(&quot;pcawg_cn_tally_X.rds&quot;)
tally_X_noLOH &lt;- readRDS(&quot;pcawg_cn_tally_X_noLOH.rds&quot;)

sigs &lt;- bp_extract_signatures(
  tally_X$nmf_matrix,
  range = 2:30,
  n_bootstrap = 10,
  n_nmf_run = 10,
  cores = 50,
  cache_dir = &quot;pcawg_bp_pynmf&quot;,
  keep_cache = TRUE,
  cores_solution = 30,
  pynmf = TRUE,
  use_conda = TRUE
)

saveRDS(sigs, file = &quot;pcawg_cn_sigs_CN176_BP.rds&quot;)</code></pre>
</div>
<div id="signature-number-determination" class="section level3">
<h3>Signature number determination</h3>
<p>The code above call 100 NMF runs (10 bootstrap catalogs and 10 NMF runs for each bootstrap catalog) for each signature number. Only 1000 NMF runs (20 bootstrap catalogs and 50 NMF runs for each bootstrap catalog) was applied for 176-component classification approach, which has the same arguments proposed by <span class="citation"><a href="#ref-degasperiPracticalFrameworkOnline2020" role="doc-biblioref">Degasperi et al.</a> (<a href="#ref-degasperiPracticalFrameworkOnline2020" role="doc-biblioref">2020</a>)</span>.</p>
<p>First, load the results.</p>
<pre class="r"><code>library(sigminer)
solution1000 &lt;- readRDS(&quot;../BP/BP_PCAWG_1000_Extraction_Result.rds&quot;)
solution100 &lt;- readRDS(&quot;../data/pcawg_cn_sigs_CN176_BP.rds&quot;)</code></pre>
<p>Next we compare some important measures for signature number determination. We can see 100 NMF runs and 1000 NMF runs can achieve very similar results.</p>
<pre class="r"><code>bp_show_survey(solution1000, add_score = F, fixed_ratio = F)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-16-1.png" width="3000" /></p>
<pre class="r"><code>bp_show_survey(solution100, add_score = F, fixed_ratio = F)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-16-2.png" width="3000" /> Basically, the 5 measures can be classified into 3 categories:</p>
<ol style="list-style-type: decimal">
<li><code>silhouette</code> and <code>similarity</code>: these two scores indicate the result stability. <em>The signature number with value decreases sharply is preferred.</em></li>
<li><code>distance</code> and <code>error</code>: these two scores indicate the difference between raw matrix and reconstructed matrix for signatures. <em>The signature number with lower value is better</em>.</li>
<li><code>pos cor</code>: this is constructed by Prof. Liu and Shixiang to see the average positive correlation between signatures. A signature number should be considered if it has lower value. In practice, we find this measure has referential value sometimes.</li>
</ol>
<p><em>More see <code>?bp_show_survey</code> in R console for how they are calculated</em>.</p>
<p>In previous study, <span class="citation"><a href="#ref-degasperiPracticalFrameworkOnline2020" role="doc-biblioref">Degasperi et al.</a> (<a href="#ref-degasperiPracticalFrameworkOnline2020" role="doc-biblioref">2020</a>)</span> used <code>silhouette</code> and <code>error</code> for signature number determination, <span class="citation"><a href="#ref-pcawgmutationalsignaturesworkinggroupRepertoireMutationalSignatures2020" role="doc-biblioref">PCAWG Mutational Signatures Working Group et al.</a> (<a href="#ref-pcawgmutationalsignaturesworkinggroupRepertoireMutationalSignatures2020" role="doc-biblioref">2020</a>)</span> used <code>similarity</code> and <code>distance</code> for signature number determination. For plots above we can see that they very close.</p>
<p><strong>From plots above, we observe that the stability drops sharply after <code>11</code>, so here <code>11</code> signatures is selected for this PCAWG dataset</strong>.</p>
</div>
<div id="similarity-between-two-results-with-different-parameter-setting" class="section level3">
<h3>Similarity between two results with different parameter setting</h3>
<p>We have two results above, so we can compare the signatures from two different parameter setting by calculating their pairwise cosine similarity.</p>
<pre class="r"><code>sim &lt;- get_sig_similarity(solution1000$object$K11, solution100$object$K11)
pheatmap::pheatmap(sim$similarity, cluster_rows = FALSE, cluster_cols = FALSE, display_numbers = TRUE)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-17-1.png" width="2400" /></p>
<p><em>We can know that the results contain same signature profile! This practice indicates that 100 NMF runs for each signature number is enough.</em></p>
<p>The <code>Signature</code> object with 11 signature profiles are stored for further analysis.</p>
<pre class="r"><code># Use root directory as work directory
saveRDS(solution1000$object$K11, file = &quot;data/pcawg_cn_sigs_CN176_signature.rds&quot;)</code></pre>
</div>
</div>
<div id="reliable-signature-activity-attribution" class="section level2">
<h2>Reliable signature activity attribution</h2>
<p>After signature determination, the reliable activities (a.k.a. exposures) of signatures are attributed to each sample for downstream analysis.</p>
<div id="attribution-pipeline" class="section level3">
<h3>Attribution pipeline</h3>
<p>There are 3 core steps in this pipeline:</p>
<ol style="list-style-type: decimal">
<li>Step1: We run <code>n = 100</code> optimizations on bootstrapped data and obtain a distribution of activities for each sample.</li>
<li>Step2: We consider the median activities as point estimates.</li>
<li>step3: In each sample, we reset a signature activity to <code>0</code> if its activity distribution below the threshold of <code>1%</code> of total mutations in the sample with one-side T test p value <code>&gt;0.05</code>.</li>
</ol>
<p>This pipeline is similar to previously reported <span class="citation"><a href="#ref-huangDetectingPresenceMutational2018" role="doc-biblioref">Huang, Wojtowicz, and Przytycka</a> (<a href="#ref-huangDetectingPresenceMutational2018" role="doc-biblioref">2018</a>)</span>. However, there are some different settings:</p>
<ol style="list-style-type: decimal">
<li>We use <code>1%</code> of total mutations as a threshold here instead <code>5%</code> in <span class="citation"><a href="#ref-degasperiPracticalFrameworkOnline2020" role="doc-biblioref">Degasperi et al.</a> (<a href="#ref-degasperiPracticalFrameworkOnline2020" role="doc-biblioref">2020</a>)</span>, because we observed some copy number segment contributions are removed from results with <code>5%</code> but they can be used for better explanation of sample copy number profile reconstruction.</li>
<li>We use T test for P value calculation here instead of empirical P value calculation based on proportion in <span class="citation"><a href="#ref-degasperiPracticalFrameworkOnline2020" role="doc-biblioref">Degasperi et al.</a> (<a href="#ref-degasperiPracticalFrameworkOnline2020" role="doc-biblioref">2020</a>)</span> for keeping more positive events.</li>
</ol>
<blockquote>
<p>TODO: add flowchart</p>
</blockquote>
</div>
<div id="apply-to-pcawg-catalog-matrix-and-signatures" class="section level3">
<h3>Apply to PCAWG catalog matrix and signatures</h3>
<p>We apply the attribution pipeline above.</p>
<pre class="r"><code># Use root directory as work directory
library(sigminer)
tally_X &lt;- readRDS(&quot;data/pcawg_cn_tally_X.rds&quot;)
pcawg_types &lt;- readRDS(&quot;data/pcawg_type_info.rds&quot;)

sc &lt;- pcawg_types$cancer_type
names(sc) &lt;- pcawg_types$sample

pcawg_expo &lt;- bp_attribute_activity(
  solution1000$object$K11,
  sample_class = sc,
  nmf_matrix = tally_X$nmf_matrix,
  return_class = &quot;data.table&quot;
)

saveRDS(pcawg_expo, file = &quot;data/pcawg_cn_sigs_CN176_activity.rds&quot;)</code></pre>
<p>Now we can know how well each sample catalog profile can be constructed from the signature combination.</p>
<pre class="r"><code>pcawg_expo &lt;- readRDS(&quot;../data/pcawg_cn_sigs_CN176_activity.rds&quot;)
summary(pcawg_expo$similarity)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.4446  0.8431  0.9043  0.8950  0.9649  0.9999 </code></pre>
<pre class="r"><code>hist(pcawg_expo$similarity, breaks = 100, xlab = &quot;Reconstructed similarity&quot;, main = NA)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-21-1.png" width="2400" /> Most of samples are well constructed. Next we will use a similarity threshold <code>0.75</code> to filter out some samples for removing their effects on the following analysis.</p>
</div>
</div>
<div id="tcga-copy-number-signatures" class="section level2">
<h2>TCGA copy number signatures</h2>
<p>To validate the copy number signatures discovered in PCAWG database, we need another database for validation. TCGA is the best resource we got and similar processing and signature identification are applied.</p>
<pre class="r"><code>library(sigminer)

tally_X &lt;- readRDS(&quot;tcga_cn_tally_X.rds&quot;)
tally_X_noLOH &lt;- readRDS(&quot;tcga_cn_tally_X_noLOH.rds&quot;)

sigs &lt;- bp_extract_signatures(
  tally_X$nmf_matrix,
  range = 2:30,
  cores = 50,
  n_bootstrap = 10,
  n_nmf_run = 10,
  cache_dir = &quot;tcga_bp_pynmf&quot;,
  keep_cache = TRUE,
  cores_solution = 30,
  pynmf = TRUE,
  use_conda = TRUE
)

saveRDS(sigs, file = &quot;tcga_cn_sigs_CN176_BP.rds&quot;)</code></pre>
<p>Load the solution list:</p>
<pre class="r"><code>tcga_solutions &lt;- readRDS(&quot;../data/TCGA/tcga_cn_sigs_CN176_BP.rds&quot;)</code></pre>
<pre class="r"><code>bp_show_survey(tcga_solutions, add_score = FALSE, fixed_ratio = FALSE, )</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-24-1.png" width="3000" /> Also <code>11</code> signatures are determined based on survey plot.</p>
<p>Similarly, we apply the attribution pipeline to TCGA.</p>
<pre class="r"><code>library(sigminer)

tcga_solutions &lt;- readRDS(&quot;data/TCGA/tcga_cn_sigs_CN176_BP.rds&quot;)

tcga_expo &lt;- bp_attribute_activity(
  tcga_solutions$object$K11,
  sample_class = sc,
  nmf_matrix = tally_X$nmf_matrix,
  return_class = &quot;data.table&quot;
)

saveRDS(tcga_expo, file = &quot;data/TCGA/tcga_cn_sigs_CN176_activity.rds&quot;)</code></pre>
<pre class="r"><code>tcga_expo &lt;- readRDS(&quot;../data/TCGA/tcga_cn_sigs_CN176_activity.rds&quot;)
summary(tcga_expo$similarity)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.2794  0.8092  0.8765  0.8623  0.9239  0.9788 </code></pre>
<pre class="r"><code>hist(tcga_expo$similarity, breaks = 100, xlab = &quot;Reconstructed similarity&quot;, main = NA)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-27-1.png" width="2400" /></p>
</div>
</div>
<div id="part-2-signature-profile-visualization-and-analysis" class="section level1">
<h1>PART 2: Signature profile visualization and analysis</h1>
<p>In this part, we would like to visualize and analyze all tumor samples as a whole.</p>
<div id="signature-profile-visualization" class="section level2">
<h2>Signature profile visualization</h2>
<pre class="r"><code>library(tidyverse)
library(sigminer)

pcawg_sigs &lt;- readRDS(file = &quot;../data/pcawg_cn_sigs_CN176_signature.rds&quot;)</code></pre>
<p>Show signature profile with relative contribution value for each component.</p>
<pre class="r"><code>show_sig_profile(
  pcawg_sigs,
  mode = &quot;copynumber&quot;,
  method = &quot;X&quot;,
  style = &quot;cosmic&quot;
)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-30-1.png" width="4200" /></p>
<p>Show signature profile with absolute contribution value (estimated segment count) for each component.</p>
<pre class="r"><code>show_sig_profile(
  pcawg_sigs,
  mode = &quot;copynumber&quot;,
  method = &quot;X&quot;,
  style = &quot;cosmic&quot;,
  normalize = &quot;raw&quot;
)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-31-1.png" width="4200" /></p>
<p>Show signature activity profile.</p>
<pre class="r"><code>show_sig_exposure(
  pcawg_sigs,
  style = &quot;cosmic&quot;,
  rm_space = TRUE,
  palette = sigminer:::letter_colors
)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-32-1.png" width="4200" /></p>
</div>
<div id="signature-analysis" class="section level2">
<h2>Signature analysis</h2>
<div id="signature-similarity" class="section level3">
<h3>Signature similarity</h3>
<p>Here we explore the similarity between copy number signatures identified in PCAWG data. High cosine similarity will cause some issues in signature assignment and activity attribution <span class="citation">(<a href="#ref-mauraPracticalGuideMutational2019" role="doc-biblioref">Maura et al. 2019</a>)</span>.</p>
<pre class="r"><code>sim &lt;- get_sig_similarity(pcawg_sigs, pcawg_sigs)</code></pre>
<pre class="r"><code>pheatmap::pheatmap(sim$similarity, cluster_cols = F, cluster_rows = F, display_numbers = TRUE)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-34-1.png" width="1650" /></p>
<p>Most of signature-pairs have similarity around <code>0.1</code>.</p>
</div>
<div id="signature-activity-distribution" class="section level3">
<h3>Signature activity distribution</h3>
<p>Transform data to long format.</p>
<pre class="r"><code>df_abs &lt;- get_sig_exposure(pcawg_sigs) %&gt;%
  tidyr::pivot_longer(cols = starts_with(&quot;Sig&quot;), names_to = &quot;sig&quot;, values_to = &quot;activity&quot;) %&gt;%
  dplyr::mutate(
    activity = log10(activity + 1),
    sig = factor(sig, levels = paste0(&quot;Sig&quot;, 1:11))
  )

df_rel &lt;- get_sig_exposure(pcawg_sigs, type = &quot;relative&quot;, rel_threshold = 0) %&gt;%
  tidyr::pivot_longer(cols = starts_with(&quot;Sig&quot;), names_to = &quot;sig&quot;, values_to = &quot;activity&quot;) %&gt;%
  dplyr::mutate(
    sig = factor(sig, levels = paste0(&quot;Sig&quot;, 1:11))
  )</code></pre>
<p>Plot with function in <code>sigminer</code>.</p>
<pre class="r"><code>show_group_distribution(
  df_abs,
  gvar = &quot;sig&quot;,
  dvar = &quot;activity&quot;,
  order_by_fun = FALSE,
  g_angle = 90,
  ylab = &quot;log10(activity+1)&quot;
)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-36-1.png" width="2400" /></p>
<pre class="r"><code>show_group_distribution(
  df_rel,
  gvar = &quot;sig&quot;,
  dvar = &quot;activity&quot;,
  order_by_fun = FALSE,
  g_angle = 90,
  ylab = &quot;relative activity&quot;
)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-37-1.png" width="2400" /></p>
</div>
</div>
<div id="copy-number-profile-and-signature-activity" class="section level2">
<h2>Copy number profile and signature activity</h2>
<div id="copy-number-profile-for-representative-samples" class="section level3">
<h3>Copy number profile for representative samples</h3>
<p>Here we will draw copy number profile of 2 representative samples for each signature.</p>
<pre class="r"><code>get_sample_from_sig &lt;- function(dt, sig) {
  res &lt;- head(dt[order(dt[[sig]], decreasing = TRUE)], 2L)
  res
}</code></pre>
<p>Read <code>CopyNumber</code> object for PCAWG data.</p>
<pre class="r"><code>pcawg_cn_obj &lt;- readRDS(&quot;../data/pcawg_cn_obj.rds&quot;)
samp_summary &lt;- pcawg_cn_obj@summary.per.sample
rel_activity &lt;- get_sig_exposure(pcawg_sigs, type = &quot;relative&quot;, rel_threshold = 0)
rel_activity &lt;- rel_activity[, lapply(.SD, function(x) {
  if (is.numeric(x)) round(x, 3) else x
})]</code></pre>
<pre class="r"><code>dir.create(&quot;../output/enrich_samples/&quot;, showWarnings = FALSE)
for (i in paste0(&quot;Sig&quot;, 1:11)) {
  cat(paste0(&quot;Most enriched in &quot;, i, &quot;\n&quot;))
  s &lt;- get_sample_from_sig(rel_activity, i)
  print(s)
  plist &lt;- show_cn_profile(pcawg_cn_obj,
    samples = s$sample,
    show_title = TRUE,
    return_plotlist = TRUE
  )
  plist &lt;- purrr::map2(plist, s$sample, function(p, x) {
    s &lt;- samp_summary[sample == x]
    text &lt;- paste0(
      &quot;n_of_seg:&quot;, s$n_of_seg, &quot;\n&quot;,
      &quot;n_of_amp:&quot;, s$n_of_amp, &quot;\n&quot;,
      &quot;n_of_del:&quot;, s$n_of_del, &quot;\n&quot;
    )
    p &lt;- p + annotate(&quot;text&quot;,
      x = Inf, y = Inf, hjust = 1, vjust = 1,
      label = text
    )
    p
  })
  p &lt;- cowplot::plot_grid(plotlist = plist, nrow = 2)
  print(p)
  ggplot2::ggsave(file.path(&quot;../output/enrich_samples/&quot;, paste0(i, &quot;.pdf&quot;)),
    plot = p, width = 12, height = 6
  )
}</code></pre>
<pre><code>Most enriched in Sig1
     sample  Sig1  Sig2  Sig3  Sig4 Sig5 Sig6  Sig7  Sig8  Sig9 Sig10 Sig11
1: SP105253 0.717 0.045 0.000 0.001    0 0.03 0.183 0.000 0.003 0.003 0.018
2:  SP98941 0.682 0.200 0.002 0.000    0 0.00 0.002 0.099 0.000 0.009 0.005</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-40-1.png" width="3600" /></p>
<pre><code>Most enriched in Sig2
    sample  Sig1  Sig2  Sig3  Sig4  Sig5  Sig6  Sig7  Sig8  Sig9 Sig10 Sig11
1: SP55235 0.056 0.796 0.000 0.006 0.007 0.031 0.010 0.066 0.000     0 0.027
2: SP56303 0.019 0.779 0.004 0.000 0.000 0.102 0.064 0.026 0.005     0 0.001</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-40-2.png" width="3600" /></p>
<pre><code>Most enriched in Sig3
     sample Sig1 Sig2  Sig3  Sig4  Sig5 Sig6  Sig7 Sig8  Sig9 Sig10 Sig11
1: SP112911    0    0 0.988 0.002 0.006    0 0.000    0 0.004     0     0
2: SP112817    0    0 0.959 0.006 0.013    0 0.005    0 0.016     0     0</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-40-3.png" width="3600" /></p>
<pre><code>Most enriched in Sig4
     sample  Sig1  Sig2  Sig3  Sig4  Sig5 Sig6  Sig7  Sig8  Sig9 Sig10 Sig11
1: SP114903 0.144 0.000 0.005 0.615 0.187    0 0.008 0.003 0.030 0.008 0.000
2: SP114898 0.166 0.001 0.005 0.594 0.194    0 0.006 0.002 0.022 0.006 0.002</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-40-4.png" width="3600" /></p>
<pre><code>Most enriched in Sig5
     sample Sig1 Sig2 Sig3 Sig4 Sig5 Sig6 Sig7 Sig8 Sig9 Sig10 Sig11
1: SP105708    0    0    0    0    1    0    0    0    0     0     0
2: SP107359    0    0    0    0    1    0    0    0    0     0     0</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-40-5.png" width="3600" /></p>
<pre><code>Most enriched in Sig6
     sample  Sig1  Sig2  Sig3  Sig4  Sig5  Sig6  Sig7  Sig8 Sig9 Sig10 Sig11
1: SP116525 0.033 0.021 0.001 0.002 0.000 0.905 0.004 0.028    0 0.003 0.004
2: SP121824 0.000 0.000 0.000 0.013 0.028 0.800 0.001 0.000    0 0.002 0.156</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-40-6.png" width="3600" /></p>
<pre><code>Most enriched in Sig7
     sample  Sig1  Sig2 Sig3 Sig4 Sig5  Sig6  Sig7  Sig8  Sig9 Sig10 Sig11
1: SP116345 0.006 0.349    0    0    0 0.003 0.591 0.051 0.000 0.000     0
2: SP124197 0.069 0.104    0    0    0 0.000 0.591 0.156 0.028 0.053     0</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-40-7.png" width="3600" /></p>
<pre><code>Most enriched in Sig8
    sample  Sig1 Sig2 Sig3 Sig4 Sig5 Sig6 Sig7  Sig8 Sig9 Sig10 Sig11
1:  SP5381 0.000    0    0    0    0    0    0 1.000    0     0     0
2: SP78664 0.014    0    0    0    0    0    0 0.986    0     0     0</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-40-8.png" width="3600" /></p>
<pre><code>Most enriched in Sig9
     sample Sig1 Sig2  Sig3  Sig4  Sig5 Sig6 Sig7 Sig8  Sig9 Sig10 Sig11
1: SP106797    0    0 0.073 0.029 0.136    0    0    0 0.762     0     0
2: SP135354    0    0 0.043 0.000 0.251    0    0    0 0.705     0     0</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-40-9.png" width="3600" /></p>
<pre><code>Most enriched in Sig10
     sample  Sig1 Sig2 Sig3  Sig4  Sig5 Sig6  Sig7  Sig8  Sig9 Sig10 Sig11
1: SP117086 0.000    0    0 0.000 0.038    0 0.065 0.094 0.043 0.760     0
2: SP117223 0.071    0    0 0.007 0.197    0 0.000 0.015 0.022 0.688     0</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-40-10.png" width="3600" /></p>
<pre><code>Most enriched in Sig11
     sample  Sig1  Sig2  Sig3  Sig4  Sig5  Sig6  Sig7  Sig8  Sig9 Sig10 Sig11
1:  SP11235 0.047 0.030 0.051 0.140 0.046 0.002 0.051 0.028 0.056 0.065 0.485
2: SP112194 0.073 0.128 0.003 0.032 0.031 0.054 0.032 0.091 0.012 0.060 0.484</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-40-11.png" width="3600" /></p>
</div>
<div id="ideograph-for-copy-number-profile-reconstruction" class="section level3">
<h3>Ideograph for copy number profile reconstruction</h3>
<blockquote>
<p>TODO: select one/two samples and draw copy number profile -&gt; catalog profile -&gt; signature relative activity. <strong>Similar to Nature paper</strong>.</p>
</blockquote>
</div>
</div>
</div>
<div id="part-3-pan-cancer-signature-analysis" class="section level1">
<h1>PART 3: Pan-Cancer signature analysis</h1>
<p>In this part, we will analyze copy number signatures across cancer types and show the landscape.</p>
<div id="signature-number-and-contribution-in-each-cancer-type" class="section level2">
<h2>Signature number and contribution in each cancer type</h2>
<p>Load tidy cancer type annotation data.</p>
<pre class="r"><code>library(sigminer)
library(tidyverse)

pcawg_types &lt;- readRDS(&quot;../data/pcawg_type_info.rds&quot;)</code></pre>
<p>To better describe the copy number signature landscape, here we use refitting activity data obtained from bootstrap procedure (see section “Reliable signature activity attribution” in PART 1).</p>
<pre class="r"><code>pcawg_activity &lt;- readRDS(&quot;../data/pcawg_cn_sigs_CN176_activity.rds&quot;)</code></pre>
<p>Combine the cancer type annotation and activity data and only keep samples with good reconstruction (<code>&gt;0.75</code> cosine similarity).</p>
<pre class="r"><code>keep_samps &lt;- pcawg_activity$similarity &gt;= 0.75

df_abs &lt;- merge(pcawg_activity$abs_activity[keep_samps], pcawg_types, by = &quot;sample&quot;)
df_rel &lt;- merge(pcawg_activity$rel_activity[keep_samps], pcawg_types, by = &quot;sample&quot;)</code></pre>
<div id="signature-activity-in-each-cancer-type" class="section level3">
<h3>Signature activity in each cancer type</h3>
<p>Here we draw distribution of a signature across cancer types.</p>
<pre class="r"><code>show_group_distribution(
  df_abs,
  gvar = &quot;cancer_type&quot;,
  dvar = &quot;Sig1&quot;,
  order_by_fun = FALSE,
  g_angle = 90,
  point_size = 0.3
)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-45-1.png" width="3600" /></p>
<p>We have many signatures here, so we output them to PDF files.</p>
<pre class="r"><code>dir.create(&quot;../output/cancer-type-dist&quot;, showWarnings = F)
signames &lt;- paste0(&quot;Sig&quot;, 1:11)
for (i in signames) {
  pxx &lt;- show_group_distribution(df_abs,
    gvar = &quot;cancer_type&quot;,
    dvar = i, order_by_fun = FALSE,
    ylab = i,
    g_angle = 90, point_size = 0.3
  )
  ggplot2::ggsave(file.path(&quot;../output/cancer-type-dist/&quot;, paste0(&quot;Absolute_activity_&quot;, i, &quot;.pdf&quot;)),
    plot = pxx, width = 12, height = 6
  )
  pxx &lt;- show_group_distribution(df_rel,
    gvar = &quot;cancer_type&quot;,
    dvar = i, order_by_fun = FALSE,
    ylab = i,
    g_angle = 90, point_size = 0.3
  )
  ggplot2::ggsave(file.path(&quot;../output/cancer-type-dist/&quot;, paste0(&quot;Relative_activity_&quot;, i, &quot;.pdf&quot;)),
    plot = pxx, width = 12, height = 6
  )
}
rm(pxx)</code></pre>
</div>
<div id="signature-landscape" class="section level3">
<h3>Signature landscape</h3>
<p>Define a signature which is detectable if this signature contribute <code>&gt;5%</code> exposures in a sample.</p>
<pre class="r"><code>df &lt;- df_rel %&gt;%
  dplyr::mutate_at(dplyr::vars(dplyr::starts_with(&quot;Sig&quot;)), ~ ifelse(. &gt; 0.05, 1L, 0L)) %&gt;%
  tidyr::pivot_longer(
    cols = dplyr::starts_with(&quot;Sig&quot;),
    names_to = &quot;sig&quot;, values_to = &quot;detectable&quot;
  )

df2 &lt;- df_rel %&gt;%
  tidyr::pivot_longer(
    cols = dplyr::starts_with(&quot;Sig&quot;),
    names_to = &quot;sig&quot;, values_to = &quot;expo&quot;
  )
df &lt;- dplyr::left_join(df, df2,
  by = c(&quot;sample&quot;, &quot;cancer_type&quot;, &quot;sig&quot;)
)

df_type &lt;- df %&gt;%
  dplyr::group_by(cancer_type, sig) %&gt;%
  dplyr::summarise(
    freq = sum(detectable), # directly use count
    expo = median(expo[detectable == 1]),
    n = n(),
    label = paste0(unique(cancer_type), &quot; (n=&quot;, n, &quot;)&quot;),
    .groups = &quot;drop&quot;
  )

mps &lt;- unique(df_type[, c(&quot;cancer_type&quot;, &quot;label&quot;)])
mpss &lt;- mps$label
names(mpss) &lt;- mps$cancer_type</code></pre>
<pre class="r"><code>summary(df_type$freq)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
   0.00    6.00   15.00   30.42   36.00  247.00 </code></pre>
<p>Show copy number signature landscape.</p>
<pre class="r"><code>library(cowplot)

p &lt;- ggplot(
  df_type,
  aes(x = cancer_type, y = factor(sig, levels = paste0(&quot;Sig&quot;, 1:11)))
) +
  geom_point(aes(size = freq, color = expo)) +
  theme_cowplot() +
  ggpubr::rotate_x_text(60) +
  scale_x_discrete(breaks = mps$cancer_type, labels = mps$label) +
  scale_size_continuous(
    limits = c(5, 250),
    breaks = c(5, 20, 50, 100, 200)
  ) +
  scale_color_stepsn(
    colors = viridis::viridis(5, direction = -1),
    breaks = c(0, 0.25, 0.5, 0.75, 1)
  ) +
  labs(
    x = NULL, y = &quot;Copy number signatures&quot;,
    color = &quot;Median activity\ndue to signature&quot;,
    size = &quot;Tumors with\nthe signature&quot;
  )
p</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-49-1.png" width="3600" /></p>
<pre class="r"><code>ggsave(&quot;../output/CNS_PCAWG_landscape.pdf&quot;,
  plot = p,
  height = 6, width = 12
)</code></pre>
</div>
</div>
<div id="cancer-type-associated-enrichment" class="section level2">
<h2>Cancer type associated enrichment</h2>
<p>Run enrichment analysis.</p>
<pre class="r"><code>enrich_result &lt;- group_enrichment(
  df_abs,
  grp_vars = &quot;cancer_type&quot;,
  enrich_vars = paste0(&quot;Sig&quot;, 1:11),
  co_method = &quot;wilcox.test&quot;
)</code></pre>
<p>Show enrichment landscape.</p>
<pre class="r"><code>enrich_result$enrich_var &lt;- factor(enrich_result$enrich_var, paste0(&quot;Sig&quot;, 1:11))
p &lt;- show_group_enrichment(enrich_result, fill_by_p_value = TRUE, return_list = T)
p &lt;- p$cancer_type + labs(x = NULL, y = NULL)
p</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-51-1.png" width="2400" /></p>
<pre class="r"><code>ggsave(&quot;../output/CNS_PCAWG_enrichment_landscape.pdf&quot;,
  plot = p,
  height = 8, width = 6
)</code></pre>
<p>To better visualize the enrichment results, we use binned color regions.</p>
<pre class="r"><code>p &lt;- show_group_enrichment(
  enrich_result,
  fill_by_p_value = TRUE,
  cut_p_value = TRUE,
  return_list = T
)
p &lt;- p$cancer_type + labs(x = NULL, y = NULL)
p</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-52-1.png" width="2400" /></p>
<pre class="r"><code>ggsave(&quot;../output/CNS_PCAWG_enrichment_landscape2.pdf&quot;,
  plot = p,
  height = 8, width = 6
)</code></pre>
<p>We see cancer type <code>SoftTissue-Liposarc</code> has pretty high enrichment on <code>Sig6</code>. Let’s check the enrichment result.</p>
<pre class="r"><code>enrich_result[grp1 == &quot;SoftTissue-Liposarc&quot;]</code></pre>
<pre><code>        grp_var enrich_var                grp1 grp2 grp1_size grp1_pos_measure
 1: cancer_type       Sig1 SoftTissue-Liposarc Rest        19       107.747229
 2: cancer_type       Sig2 SoftTissue-Liposarc Rest        19        45.210722
 3: cancer_type       Sig3 SoftTissue-Liposarc Rest        19        17.841799
 4: cancer_type       Sig4 SoftTissue-Liposarc Rest        19        53.524895
 5: cancer_type       Sig5 SoftTissue-Liposarc Rest        19         9.332606
    grp2_size grp2_pos_measure measure_observed measure_tested      p_value
 1:      2621        17.690767        6.0905910             NA 2.618595e-02
 2:      2621        14.791334        3.0565682             NA 5.269566e-01
 3:      2621        15.222509        1.1720669             NA 7.519819e-02
 4:      2621        13.483545        3.9696455             NA 1.051345e-03
 5:      2621        14.027543        0.6653058             NA 2.968415e-02
          type      method
 1: continuous wilcox.test
 2: continuous wilcox.test
 3: continuous wilcox.test
 4: continuous wilcox.test
 5: continuous wilcox.test
 [ reached getOption(&quot;max.print&quot;) -- omitted 6 rows ]</code></pre>
<p>We can see mean activity <code>486</code> (<code>n=19</code>) vs <code>9</code> (<code>n=2621</code>).</p>
<p>Let’s go further plot the distribution for the two groups.</p>
<pre class="r"><code>df_check &lt;- df_abs[, c(&quot;Sig6&quot;, &quot;cancer_type&quot;)][
  , .(
    cancer_type = ifelse(cancer_type == &quot;SoftTissue-Liposarc&quot;,
      &quot;SoftTissue-Liposarc&quot;,
      &quot;Others&quot;
    ),
    Sig6 = Sig6
  )
]</code></pre>
<pre class="r"><code>ggpubr::ggboxplot(
  df_check,
  x = &quot;cancer_type&quot;, y = &quot;Sig6&quot;,
  fill = &quot;cancer_type&quot;,
  xlab = FALSE, width = 0.3
)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-55-1.png" width="2400" /> Check copy number distribution for the <code>"SoftTissue-Liposarc"</code> samples.</p>
<pre class="r"><code>samples &lt;- df_abs[cancer_type == &quot;SoftTissue-Liposarc&quot;]$sample

pcawg_cn_obj &lt;- readRDS(&quot;../data/pcawg_cn_obj.rds&quot;)
cn_dt &lt;- subset(pcawg_cn_obj@data, sample %in% samples)
cn_dt$segLen &lt;- cn_dt$end - cn_dt$start + 1</code></pre>
<p>Copy number value:</p>
<pre class="r"><code>boxplot(cn_dt$segVal)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-57-1.png" width="2400" /></p>
<p>Segment length:</p>
<pre class="r"><code>boxplot(cn_dt$segLen)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-58-1.png" width="2400" /></p>
<pre class="r"><code>cn_dt_samp &lt;- cn_dt[, .(nAMP = sum(segVal &gt; 2)), by = sample]</code></pre>
<pre class="r"><code>boxplot(cn_dt_samp$nAMP)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-60-1.png" width="2400" /></p>
</div>
</div>
<div id="part-4-cancer-subtyping-and-prognosis-analysis" class="section level1">
<h1>PART 4: Cancer subtyping and prognosis analysis</h1>
<p>In this part, we would like to cluster all tumors based on CN signatures and use the clusters to explore Pan-Cancer heterogeneity.</p>
<div id="data-integration" class="section level2">
<h2>Data integration</h2>
<p>Many raw data have been processed by tools (or collected) and cleaned. This part I mainly describe how to combine them to construct the integrated dataset.</p>
<blockquote>
<p>If you care about the data (source) shown below, please check the last section of <strong>PART 0</strong>.</p>
</blockquote>
<pre class="r"><code>library(pacman)
p_load(
  tidyverse,
  data.table,
  sigminer,
  IDConverter
)</code></pre>
<p>Load pre-processed data.</p>
<pre class="r"><code>pcawg_cn_obj &lt;- readRDS(&quot;../data/pcawg_cn_obj.rds&quot;)
pcawg_cn_sig &lt;- readRDS(&quot;../data/pcawg_cn_sigs_CN176_signature.rds&quot;)
pcawg_samp_info_sp &lt;- readRDS(&quot;../data/pcawg_samp_info_sp.rds&quot;)
samp_summary &lt;- pcawg_cn_obj@summary.per.sample
samp_summary$n_loh &lt;- NULL</code></pre>
<p>Check if all samples are recorded.</p>
<pre class="r"><code>all(pcawg_cn_obj@summary.per.sample$sample %in% pcawg_samp_info_sp$pcawg_specimen_histology_August2016_v9$icgc_specimen_id)</code></pre>
<pre class="r"><code>pcawg_samp_info_sp &lt;- lapply(pcawg_samp_info_sp, function(x) {
  colnames(x)[1] &lt;- &quot;sample&quot;
  x
})
pcawg_samp_info &lt;- purrr::reduce(pcawg_samp_info_sp, dplyr::full_join, by = &quot;sample&quot;)
pcawg_samp_info &lt;- pcawg_samp_info %&gt;% dplyr::filter(sample %in% pcawg_cn_obj@summary.per.sample$sample)

rm(pcawg_samp_info_sp)</code></pre>
<p>Refer to <span class="citation"><a href="#ref-pcawgmutationalsignaturesworkinggroupRepertoireMutationalSignatures2020" role="doc-biblioref">PCAWG Mutational Signatures Working Group et al.</a> (<a href="#ref-pcawgmutationalsignaturesworkinggroupRepertoireMutationalSignatures2020" role="doc-biblioref">2020</a>)</span>, some cancer types with small sample size should be combined.</p>
<pre class="r"><code>table(pcawg_samp_info$histology_abbreviation)</code></pre>
<pre class="r"><code>pcawg_samp_info &lt;- pcawg_samp_info %&gt;%
  mutate(
    cancer_type = case_when(
      startsWith(histology_abbreviation, &quot;Breast&quot;) ~ &quot;Breast&quot;,
      startsWith(histology_abbreviation, &quot;Biliary-AdenoCA&quot;) ~ &quot;Biliary-AdenoCA&quot;,
      histology_abbreviation %in% c(&quot;Bone-Epith&quot;, &quot;Bone-Benign&quot;) ~ &quot;Bone-Other&quot;,
      startsWith(histology_abbreviation, &quot;Cervix&quot;) ~ &quot;Cervix&quot;,
      histology_abbreviation %in% c(&quot;Myeloid-AML, Myeloid-MDS&quot;, &quot;Myeloid-AML, Myeloid-MPN&quot;, &quot;Myeloid-MDS&quot;, &quot;Myeloid-MPN&quot;) ~ &quot;Myeloid-MDS/MPN&quot;,
      TRUE ~ histology_abbreviation
    )
  )</code></pre>
<p>Select important features.</p>
<pre class="r"><code>pcawg_samp_info &lt;- pcawg_samp_info %&gt;%
  dplyr::select(c(
    &quot;sample&quot;, &quot;_PATIENT&quot;, &quot;donor_sex&quot;, &quot;donor_age_at_diagnosis&quot;,
    &quot;donor_survival_time&quot;, &quot;donor_vital_status&quot;, &quot;first_therapy_type&quot;,
    &quot;tobacco_smoking_history_indicator&quot;, &quot;alcohol_history&quot;,
    &quot;dcc_project_code&quot;, &quot;dcc_specimen_type&quot;,
    &quot;histology_abbreviation&quot;, &quot;cancer_type&quot;,
    &quot;tumour_stage&quot;,
    &quot;level_of_cellularity&quot;
  ))
colnames(pcawg_samp_info)[2] &lt;- &quot;donor_id&quot;

# Remove duplicated samples
pcawg_samp_info &lt;- pcawg_samp_info[!duplicated(pcawg_samp_info$sample), ]</code></pre>
<p>Get signature activities.</p>
<pre class="r"><code>pcawg_activity &lt;- readRDS(&quot;../data/pcawg_cn_sigs_CN176_activity.rds&quot;)

expo_abs &lt;- pcawg_activity$abs_activity[, lapply(.SD, function(x) if (is.numeric(x)) round(x, digits = 3) else x)]
colnames(expo_abs) &lt;- c(&quot;sample&quot;, paste0(&quot;Abs_&quot;, colnames(expo_abs)[-1]))


expo_rel &lt;- pcawg_activity$rel_activity[, lapply(.SD, function(x) if (is.numeric(x)) round(x, digits = 3) else x)]
colnames(expo_rel) &lt;- c(&quot;sample&quot;, paste0(&quot;Rel_&quot;, colnames(expo_rel)[-1]))

pcawg_activity2 &lt;- dplyr::left_join(expo_abs, expo_rel, by = &quot;sample&quot;)
pcawg_activity2$similarity &lt;- pcawg_activity$similarity
pcawg_activity &lt;- pcawg_activity2
pcawg_activity$keep &lt;- pcawg_activity$similarity &gt;= 0.75

rm(pcawg_activity2, expo_abs, expo_rel)</code></pre>
<p>Read tumor purity &amp; ploidy &amp; WGD status.</p>
<pre class="r"><code>dt_pp &lt;- fread(&quot;../data/PCAWG/consensus.20170217.purity.ploidy_sp&quot;)
dt_pp &lt;- dt_pp[, c(&quot;samplename&quot;, &quot;purity&quot;, &quot;ploidy&quot;, &quot;wgd_status&quot;)]
colnames(dt_pp)[1] &lt;- &quot;sample&quot;</code></pre>
<p>Calculate fusion events in all samples.</p>
<pre class="r"><code>fusion &lt;- fread(&quot;../data/PCAWG/pcawg3_fusions_PKU_EBI.gene_centric.sp.xena&quot;)
fusion &lt;- as.matrix(fusion[, -1])
fusion &lt;- apply(fusion, 2, sum)
fusion &lt;- dplyr::tibble(
  sample = names(fusion),
  n_fusion = as.numeric(fusion)
)</code></pre>
<p>Load HRD status in all samples.</p>
<pre class="r"><code>HRD &lt;- readRDS(&quot;../data/PCAWG/pcawg_CHORD.rds&quot;)
HRD &lt;- HRD %&gt;%
  dplyr::select(icgc_specimen_id, is_hrd, genotype_monoall) %&gt;%
  set_names(c(&quot;sample&quot;, &quot;hrd_status&quot;, &quot;hrd_genotype&quot;)) %&gt;%
  unique()</code></pre>
<p>Read LOH data.</p>
<pre class="r"><code>LOH &lt;- readRDS(&quot;../data/pcawg_loh.rds&quot;)</code></pre>
<p>Read Chromothripsis data and clean it.</p>
<pre class="r"><code># http://compbio.med.harvard.edu/chromothripsis/
chromothripsis &lt;- fread(&quot;../data/PCAWG/PCAWG-ShatterSeek.txt.gz&quot;)
chromothripsis &lt;- chromothripsis[, c(&quot;icgc_donor_id&quot;, &quot;type_chromothripsis&quot;, &quot;chromo_label&quot;)]

# Check label
table(chromothripsis$chromo_label)
any(is.na(chromothripsis$chromo_label))

# Summarize
chromothripsis_summary &lt;- chromothripsis[
  , .(
    n_chromo = sum(chromo_label != &quot;No&quot;),
    chromo_type = paste0(na.omit(type_chromothripsis), collapse = &quot;/&quot;)
  ),
  by = icgc_donor_id
]

chromothripsis_summary &lt;- merge(chromothripsis_summary, pcawg_full[, c(&quot;icgc_donor_id&quot;, &quot;icgc_specimen_id&quot;)],
  by = &quot;icgc_donor_id&quot;
) %&gt;% unique()
chromothripsis_summary$icgc_donor_id &lt;- NULL
colnames(chromothripsis_summary)[3] &lt;- &quot;sample&quot;
rm(chromothripsis)</code></pre>
<p>Read AmpliconArchitect results.</p>
<pre class="r"><code>aa1 &lt;- readxl::read_excel(&quot;../data/PCAWG/ecDNA.xlsx&quot;, skip = 1)
aa2 &lt;- readxl::read_excel(&quot;../data/PCAWG/ecDNA.xlsx&quot;, sheet = 3)

table(aa1$amplicon_classification)
table(aa2$sample_classification)
all(aa1$sample_barcode %in% aa2$sample_barcode)

aa_summary &lt;- aa1 %&gt;%
  dplyr::group_by(sample_barcode) %&gt;%
  dplyr::summarise(
    n_amplicon_BFB = sum(amplicon_classification == &quot;BFB&quot;, na.rm = TRUE),
    n_amplicon_ecDNA = sum(amplicon_classification == &quot;Circular&quot;, na.rm = TRUE),
    n_amplicon_HR = sum(amplicon_classification == &quot;Heavily-rearranged&quot;, na.rm = TRUE),
    n_amplicon_Linear = sum(amplicon_classification == &quot;Linear&quot;, na.rm = TRUE),
    .groups = &quot;drop&quot;
  )

data.table::setDT(aa_summary)

custom_dt &lt;- IDConverter::pcawg_simple[, c(&quot;submitted_specimen_id&quot;, &quot;icgc_specimen_id&quot;)]
custom_dt &lt;- custom_dt[startsWith(submitted_specimen_id, &quot;TCGA&quot;)]
custom_dt[, submitted_specimen_id := substr(submitted_specimen_id, 1, 15)]

aa_summary[
  , sample := ifelse(
    startsWith(sample_barcode, &quot;SA&quot;),
    convert_pcawg(sample_barcode, from = &quot;icgc_sample_id&quot;, to = &quot;icgc_specimen_id&quot;),
    convert_custom(sample_barcode, from = &quot;submitted_specimen_id&quot;, to = &quot;icgc_specimen_id&quot;, dt = custom_dt)
  )
]

## Filter out samples not in PCAWG
aa_summary &lt;- aa_summary[!is.na(sample)][, sample_barcode := NULL]

rm(aa1, aa2, custom_dt)</code></pre>
<p>Read TelomereHunter results.</p>
<pre class="r"><code>TC &lt;- readxl::read_excel(&quot;../data/PCAWG/TelomereContent.xlsx&quot;)
TC &lt;- TC %&gt;% dplyr::select(c(&quot;icgc_specimen_id&quot;, &quot;tel_content_log2&quot;))
colnames(TC)[1] &lt;- &quot;sample&quot;
data.table::setDT(TC)</code></pre>
<pre class="r"><code>apobec &lt;- read_tsv(&quot;../data/PCAWG/MAF_Aug31_2016_sorted_A3A_A3B_comparePlus.txt&quot;, col_types = cols())
apobec &lt;- apobec %&gt;%
  dplyr::select(c(1, 5)) %&gt;%
  data.table::as.data.table() %&gt;%
  setNames(c(&quot;sample&quot;, &quot;APOBEC_mutations&quot;))</code></pre>
<p>Combine and save result combined data.</p>
<pre class="r"><code>pcawg_samp_info &lt;- purrr::reduce(list(
  pcawg_samp_info,
  pcawg_activity,
  samp_summary,
  dt_pp,
  fusion,
  HRD,
  LOH,
  chromothripsis_summary,
  aa_summary,
  TC,
  apobec
),
dplyr::left_join,
by = &quot;sample&quot;
)

saveRDS(pcawg_samp_info, file = &quot;../data/pcawg_sample_tidy_info.rds&quot;)

## Remove all objects
rm(list = ls())</code></pre>
</div>
<div id="clustering-with-signature-activity" class="section level2">
<h2>Clustering with signature activity</h2>
<p>Here, we use recent consensus clustering toolkit <a href="https://github.com/jokergoo/cola">cola</a>.</p>
<p>We take 2 steps to obtain a robust clustering result:</p>
<ol style="list-style-type: decimal">
<li>We sort all samples by their total signature activities and randomly select <code>500</code> samples for running multiple methods provided by <code>cola</code> at the same time, then pick up the optimal method combination.</li>
<li>We run clustering for all samples by the method combination above and check the cola report to determine the suitable cluster number.</li>
</ol>
<div id="step-1-select-suitable-method-combination" class="section level3">
<h3>Step 1: select suitable method combination</h3>
<p>There are 2 key arguments in <code>cola</code> clustering function:</p>
<ol style="list-style-type: decimal">
<li><code>top_value_method</code>: used to extract rows (i.e. signatures here) with top values.</li>
<li><code>partition_method</code>: used to select partition method.</li>
</ol>
<p>We try all combinations with randomly selected <code>500</code> samples to explore the suitable setting.</p>
<pre class="r"><code>library(cola)
library(tidyverse)

df &lt;- readRDS(&quot;../data/pcawg_sample_tidy_info.rds&quot;)

mat_abs &lt;- df %&gt;%
  filter(keep) %&gt;%
  select(sample, starts_with(&quot;Abs_Sig&quot;)) %&gt;%
  column_to_rownames(&quot;sample&quot;) %&gt;%
  t()
mat_abs &lt;- adjust_matrix(mat_abs)
rownames(mat_abs)


# Select suitable parameters ----------------------------------------------

ds &lt;- colSums(mat_abs)
boxplot(ds)

set.seed(123)
select_samps &lt;- sample(names(sort(ds)), 500)

boxplot(ds[select_samps])

rl_abs &lt;- run_all_consensus_partition_methods(mat_abs[, select_samps], top_n = 10, mc.cores = 8, max_k = 10)
cola_report(rl_abs, output_dir = &quot;../output/cola_report/pcawg_abs_sigs_500_sampls&quot;, mc.cores = 8)</code></pre>
<p>The <code>cola</code> output report is very big and isn’t suitable to show, here I only include key figures to determine the parameter setting.</p>
<pre class="r"><code># Cluster number 2
knitr::include_graphics(&quot;cola-test-figs/tab-collect-stats-from-consensus-partition-list-1-1.png&quot;)</code></pre>
<p><img src="cola-test-figs/tab-collect-stats-from-consensus-partition-list-1-1.png" width="100%" /></p>
<pre class="r"><code># Cluster number 3
knitr::include_graphics(&quot;cola-test-figs/tab-collect-stats-from-consensus-partition-list-2-1.png&quot;)</code></pre>
<p><img src="cola-test-figs/tab-collect-stats-from-consensus-partition-list-2-1.png" width="100%" /></p>
<pre class="r"><code># Cluster number 4
knitr::include_graphics(&quot;cola-test-figs/tab-collect-stats-from-consensus-partition-list-3-1.png&quot;)</code></pre>
<p><img src="cola-test-figs/tab-collect-stats-from-consensus-partition-list-3-1.png" width="100%" /></p>
<pre class="r"><code># Cluster number 5
knitr::include_graphics(&quot;cola-test-figs/tab-collect-stats-from-consensus-partition-list-4-1.png&quot;)</code></pre>
<p><img src="cola-test-figs/tab-collect-stats-from-consensus-partition-list-4-1.png" width="100%" /></p>
<pre class="r"><code># Cluster number 6
knitr::include_graphics(&quot;cola-test-figs/tab-collect-stats-from-consensus-partition-list-5-1.png&quot;)</code></pre>
<p><img src="cola-test-figs/tab-collect-stats-from-consensus-partition-list-5-1.png" width="100%" /></p>
<pre class="r"><code># Cluster number 7
knitr::include_graphics(&quot;cola-test-figs/tab-collect-stats-from-consensus-partition-list-6-1.png&quot;)</code></pre>
<p><img src="cola-test-figs/tab-collect-stats-from-consensus-partition-list-6-1.png" width="100%" /></p>
<pre class="r"><code># Cluster number 8
knitr::include_graphics(&quot;cola-test-figs/tab-collect-stats-from-consensus-partition-list-7-1.png&quot;)</code></pre>
<p><img src="cola-test-figs/tab-collect-stats-from-consensus-partition-list-7-1.png" width="100%" /></p>
<pre class="r"><code># Cluster number 9
knitr::include_graphics(&quot;cola-test-figs/tab-collect-stats-from-consensus-partition-list-8-1.png&quot;)</code></pre>
<p><img src="cola-test-figs/tab-collect-stats-from-consensus-partition-list-8-1.png" width="100%" /></p>
<pre class="r"><code># Cluster number 10
knitr::include_graphics(&quot;cola-test-figs/tab-collect-stats-from-consensus-partition-list-9-1.png&quot;)</code></pre>
<p><img src="cola-test-figs/tab-collect-stats-from-consensus-partition-list-9-1.png" width="100%" /></p>
</div>
<div id="step-2-select-suitable-cluster-number" class="section level3">
<h3>Step 2: select suitable cluster number</h3>
<p>From figures above we do clearly see that the <code>skmeans</code> partition method fits our data and any <code>top_value_method</code> is okay. Finally, we choose combine <code>skmeans</code> (partition method) and <code>ATC</code> (top value method introduced firstly by <code>cola</code>) to run clustering for all samples.</p>
<pre class="r"><code>final_abs &lt;- run_all_consensus_partition_methods(
  mat_abs,
  top_value_method = &quot;ATC&quot;,
  partition_method = &quot;skmeans&quot;,
  top_n = 10, mc.cores = 8, max_k = 10
)
cola_report(final_abs, output_dir = &quot;../output/cola_report/pcawg_abs_sigs_all_sampls_final&quot;, mc.cores = 8)</code></pre>
<p>The report can be viewed at <a href="./pcawg_abs_sigs_all_sampls_final/cola_report.html">here</a>.</p>
<p>Next, we will load the data and visualize to select suitable cluster number.</p>
<pre class="r"><code>library(cola)
library(ComplexHeatmap)
library(tidyverse)

cluster_res &lt;- readRDS(&quot;../data/pcawg_cola_result.rds&quot;)
cluster_res &lt;- cluster_res[&quot;ATC&quot;, &quot;skmeans&quot;]
cluster_res</code></pre>
<pre><code>A &#39;ConsensusPartition&#39; object with k = 2, 3, 4, 5, 6, 7, 8, 9, 10.
  On a matrix with 10 rows and 2640 columns.
  Top rows (10) are extracted by &#39;ATC&#39; method.
  Subgroups are detected by &#39;skmeans&#39; method.
  Performed in total 450 partitions by row resampling.
  Best k for subgroups seems to be 2.

Following methods can be applied to this &#39;ConsensusPartition&#39; object:
 [1] &quot;cola_report&quot;             &quot;collect_classes&quot;        
 [3] &quot;collect_plots&quot;           &quot;collect_stats&quot;          
 [5] &quot;colnames&quot;                &quot;compare_signatures&quot;     
 [7] &quot;consensus_heatmap&quot;       &quot;dimension_reduction&quot;    
 [9] &quot;functional_enrichment&quot;   &quot;get_anno_col&quot;           
[11] &quot;get_anno&quot;                &quot;get_classes&quot;            
[13] &quot;get_consensus&quot;           &quot;get_matrix&quot;             
[15] &quot;get_membership&quot;          &quot;get_param&quot;              
[17] &quot;get_signatures&quot;          &quot;get_stats&quot;              
[19] &quot;is_best_k&quot;               &quot;is_stable_k&quot;            
[21] &quot;membership_heatmap&quot;      &quot;ncol&quot;                   
[23] &quot;nrow&quot;                    &quot;plot_ecdf&quot;              
[25] &quot;predict_classes&quot;         &quot;rownames&quot;               
[27] &quot;select_partition_number&quot; &quot;show&quot;                   
[29] &quot;suggest_best_k&quot;          &quot;test_to_known_factors&quot;  </code></pre>
<p>Show different statistics along with different cluster number <code>k</code>, which helps to determine the “best k.”</p>
<pre class="r"><code>select_partition_number(cluster_res)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-84-1.png" width="2400" /> There are many details about all statistical measures shown above at <a href="http://bioconductor.org/packages/devel/bioc/vignettes/cola/inst/doc/cola.html#toc_13">cola vignette</a>. The best k <code>2</code> is suggested by <code>cola</code> package due to its better statistical measures. However, in this practice, I think <code>3</code> or <code>6</code> is more suitable. I have the following strong reasons.</p>
<p>The reasons for <code>k = 3</code>:</p>
<ol style="list-style-type: decimal">
<li>Measure <code>1 - PAC</code> reaches a local maximal at <code>k = 3</code> which indicates the subgroups are less due to ambiguously assigned.</li>
<li>Measure <code>area_increased</code> drops sharply at <code>k = 3</code> which indicates increasing <code>k</code> won’t change the consensus matrix too much.</li>
<li>Measure <code>Jaccard</code> increases significantly at <code>k = 3</code> which indicate the subgroups at <code>k = 4</code> are highly similar as <code>k = 3</code>.</li>
</ol>
<p>The reasons for <code>k = 6</code>:</p>
<ol style="list-style-type: decimal">
<li>Measure <code>1 - PAC</code> reaches top value at <code>k = 6</code>.</li>
<li>Measure <code>Jaccard</code> increases significantly at <code>k = 6</code> which indicate the subgroups at <code>k = 7</code> are highly similar as <code>k = 6</code>.</li>
<li>Measure <code>Silhouette</code> start dropping at <code>k = 6</code>.</li>
</ol>
<p>I will draw another plot to show why the two <code>k</code>s are suitable.</p>
<pre class="r"><code>collect_classes(cluster_res)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-85-1.png" width="3900" /> We can see that subgroups for <code>k = 3, 4, 5</code> are similar and subgroups for <code>k = 6, 7, 8, 9, 10</code> are similar.</p>
<p>Draw all plots into one single page.</p>
<pre class="r"><code>pdf(file = &quot;../output/cola_all_plots.pdf&quot;, width = 16, height = 10)
collect_plots(cluster_res)
dev.off()</code></pre>
<p>Now we get the cluster labels of samples for downstream analysis.</p>
<pre class="r"><code>pcawg_clusters &lt;- get_classes(cluster_res, k = c(3, 6))
saveRDS(pcawg_clusters, file = &quot;../data/pcawg_clusters.rds&quot;)</code></pre>
</div>
</div>
<div id="heatmap-for-clustering-and-genotypephenotype-features" class="section level2">
<h2>Heatmap for clustering and genotype/phenotype features</h2>
<p>Load tidy sample info.</p>
<pre class="r"><code>tidy_info &lt;- readRDS(&quot;../data/pcawg_sample_tidy_info.rds&quot;)</code></pre>
<p>Clean some info.</p>
<pre class="r"><code>detect_any &lt;- function(x, p) {
  if (length(p) == 1L) {
    y &lt;- stringr::str_detect(x, p)
  } else {
    y &lt;- purrr::map(p, ~ stringr::str_detect(x, .))
    y &lt;- purrr::reduce(y, `|`)
  }
  y[is.na(y)] &lt;- FALSE
  y
}

tidy_info &lt;- tidy_info %&gt;%
  dplyr::mutate(
    # Roughly reassign the staging to TNM stages
    # Basically follow 7th TNM staging version, see plot &lt;https://bkimg.cdn.bcebos.com/pic/a8014c086e061d952019ec8773f40ad162d9ca36?x-bce-process=image/watermark,image_d2F0ZXIvYmFpa2U4MA==,g_7,xp_5,yp_5&gt;
    #
    # Other references:
    # https://www.cancer.org/treatment/understanding-your-diagnosis/staging.html
    # https://www.cancer.gov/publications/dictionaries/cancer-terms/def/abcd-rating
    # https://web.archive.org/web/20081004121942/http://www.oncologychannel.com/prostatecancer/stagingsystems.shtml
    # https://baike.baidu.com/item/TNM%E5%88%86%E6%9C%9F%E7%B3%BB%E7%BB%9F/10700513
    tumour_stage = case_when(
      tumour_stage %in% c(&quot;1&quot;, &quot;1a&quot;, &quot;1b&quot;, &quot;A&quot;, &quot;I&quot;, &quot;IA&quot;, &quot;IB&quot;, &quot;T1c&quot;, &quot;T1N0&quot;) | detect_any(tumour_stage, c(&quot;T1N0M0&quot;, &quot;T1aN0M0&quot;, &quot;T1bN0M0&quot;, &quot;T2aN0M0&quot;)) ~ &quot;I&quot;,
      tumour_stage %in% c(&quot;2&quot;, &quot;2a&quot;, &quot;2b&quot;, &quot;B&quot;, &quot;II&quot;, &quot;IIA&quot;, &quot;IIB&quot;, &quot;T3a&quot;, &quot;T3aN0&quot;) | detect_any(tumour_stage, c(&quot;T2bN0M0&quot;, &quot;T3N0M0&quot;, &quot;T[^34]N1.*M0&quot;)) ~ &quot;II&quot;,
      tumour_stage %in% c(&quot;3&quot;, &quot;3a&quot;, &quot;3b&quot;, &quot;3c&quot;, &quot;C&quot;, &quot;III&quot;, &quot;IIIA&quot;, &quot;IIIB&quot;, &quot;IIIC&quot;) | detect_any(tumour_stage, c(&quot;N[23].*M0&quot;, &quot;T3N1.*M0&quot;, &quot;T4.*M0&quot;)) ~ &quot;III&quot;,
      tumour_stage %in% c(&quot;4&quot;, &quot;IV&quot;, &quot;IVA&quot;) | detect_any(tumour_stage, &quot;M[^0X]&quot;) ~ &quot;IV&quot;,
      TRUE ~ &quot;Unknown&quot;
    ),
    first_therapy_type = ifelse(first_therapy_type == &quot;monoclonal antibodies (for liquid tumours)&quot;,
      &quot;other therapy&quot;, first_therapy_type
    )
  )</code></pre>
<p>Generate annotation data for plotting.</p>
<pre class="r"><code>anno_df &lt;- tidy_info %&gt;%
  dplyr::filter(keep) %&gt;%
  dplyr::select(
    sample,
    donor_age_at_diagnosis, n_of_amp, n_of_del, n_of_vchr, cna_burden, purity, ploidy,
    n_LOH, n_fusion, n_chromo, n_amplicon_BFB, n_amplicon_ecDNA, n_amplicon_HR, n_amplicon_Linear,
    tel_content_log2,
    donor_sex, tumour_stage, wgd_status, hrd_status
  ) %&gt;%
  dplyr::mutate_at(
    vars(n_fusion, n_chromo, n_amplicon_BFB, n_amplicon_ecDNA, n_amplicon_HR, n_amplicon_Linear),
    ~ ifelse(is.na(.), 0, .)
  ) %&gt;%
  dplyr::mutate(hrd_status = ifelse(hrd_status, &quot;hrd&quot;, &quot;no_hrd&quot;)) %&gt;%
  rename(
    Age = donor_age_at_diagnosis,
    AMPs = n_of_amp,
    DELs = n_of_del,
    Affected_Chrs = n_of_vchr,
    CNA_Burden = cna_burden,
    Purity = purity,
    Ploidy = ploidy,
    LOHs = n_LOH,
    Fusions = n_fusion,
    Chromothripsis = n_chromo,
    BFBs = n_amplicon_BFB,
    ecDNAs = n_amplicon_ecDNA,
    HRs = n_amplicon_HR,
    Linear_Amplicons = n_amplicon_Linear,
    `Telomere_Content(log2)` = tel_content_log2,
    Sex = donor_sex,
    Tumor_Stage = tumour_stage,
    WGD_Status = wgd_status,
    HRD_Status = hrd_status
  ) %&gt;%
  tibble::column_to_rownames(&quot;sample&quot;)

saveRDS(anno_df, file = &quot;../data/pcawg_tidy_anno_df.rds&quot;)

left_annotation &lt;- rowAnnotation(foo = anno_text(paste0(&quot;Sig&quot;, c(1:4, 6:11))))</code></pre>
<p>Show heatmap for clusters based on signature activities.</p>
<pre class="r"><code>get_signatures(cluster_res,
  k = 6, silhouette_cutoff = 0.2, anno = anno_df,
  left_annotation = left_annotation
)</code></pre>
<pre><code>* 2522/2640 samples (in 6 classes) remain after filtering by silhouette (&gt;= 0.2).
* cache hash: adc95c3acc3e95ec20e2157e35327ca3 (seed 888).
* calculating row difference between subgroups by Ftest.
* 10 signatures (100.0%) under fdr &lt; 0.05, group_diff &gt; 0.
* making heatmaps for signatures.</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-91-1.png" width="6600" /></p>
<p>Save to file.</p>
<pre class="r"><code>pdf(file = &quot;../output/pcawg_cluster_heatmap.pdf&quot;, width = 22, height = 10, onefile = FALSE)
get_signatures(cluster_res, k = 6, silhouette_cutoff = 0.2, anno = anno_df, left_annotation = left_annotation)
dev.off()</code></pre>
<p>Considering the results of <code>6</code> clusters gives confident labels for <code>&gt;90%</code> samples and thus there is no need to use <code>3</code> clusters in this study.</p>
</div>
<div id="exploration-of-patients-prognosis-difference-across-clusters" class="section level2">
<h2>Exploration of patients’ prognosis difference across clusters</h2>
<div id="clean-data-for-survival-analysis" class="section level3">
<h3>Clean data for survival analysis</h3>
<pre class="r"><code>library(ezcox)
library(tidyverse)

cluster_df &lt;- readRDS(&quot;../data/pcawg_clusters.rds&quot;) %&gt;%
  as.data.frame() %&gt;%
  tibble::rownames_to_column(&quot;sample&quot;)

tidy_info &lt;- readRDS(&quot;../data/pcawg_sample_tidy_info.rds&quot;)

df_os &lt;- tidy_info %&gt;%
  dplyr::filter(keep) %&gt;%
  dplyr::select(sample, donor_vital_status, donor_survival_time) %&gt;%
  purrr::set_names(c(&quot;sample&quot;, &quot;os&quot;, &quot;time&quot;)) %&gt;%
  na.omit() %&gt;%
  mutate(os = ifelse(os == &quot;deceased&quot;, 1, 0)) %&gt;%
  left_join(cluster_df, by = &quot;sample&quot;)

colnames(df_os)[5] &lt;- &quot;cluster&quot;
# Use the group with minimal CNV level as reference group
df_os$cluster &lt;- paste0(&quot;subgroup&quot;, df_os$cluster)
df_os$cluster &lt;- factor(df_os$cluster, levels = paste0(&quot;subgroup&quot;, c(2, 1, 3:6)))</code></pre>
<p>Available sample number with OS data:</p>
<pre class="r"><code>nrow(df_os)</code></pre>
<pre><code>[1] 1669</code></pre>
</div>
<div id="forest-plot" class="section level3">
<h3>Forest plot</h3>
<p>It’s pretty easy to visualize the relationship between subgroups and patient’s prognosis with forest plot by R package <code>ezcox</code> developed by me before.</p>
<pre class="r"><code>show_forest(df_os, covariates = &quot;cluster&quot;, status = &quot;os&quot;, add_caption = FALSE, merge_models = TRUE)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-95-1.png" width="2400" /></p>
</div>
<div id="k-m-plot" class="section level3">
<h3>K-M plot</h3>
<p>We can also check the OS difference with K-M plot.</p>
<pre class="r"><code>library(survival)
library(survminer)</code></pre>
<pre class="r"><code>sfit &lt;- survfit(Surv(time, os) ~ cluster, data = df_os)
ggsurvplot(sfit, pval = TRUE)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-97-1.png" width="2400" /></p>
<p>The result has similar meaning as forest plot.</p>
</div>
</div>
<div id="the-difference-of-genotypephenotype-measures-across-clusters" class="section level2">
<h2>The difference of genotype/phenotype measures across clusters</h2>
<div id="signatures" class="section level3">
<h3>Signatures</h3>
<p>Load SBS/DBS/ID signature activities generated from PCAWG Nature Study.</p>
<pre class="r"><code>pcawg_sbs &lt;- read_csv(&quot;../data/PCAWG/PCAWG_sigProfiler_SBS_signatures_in_samples.csv&quot;)
pcawg_dbs &lt;- read_csv(&quot;../data/PCAWG/PCAWG_sigProfiler_DBS_signatures_in_samples.csv&quot;)
pcawg_id &lt;- read_csv(&quot;../data/PCAWG/PCAWG_SigProfiler_ID_signatures_in_samples.csv&quot;)</code></pre>
<p>Merge data.</p>
<pre class="r"><code>df_merged &lt;- purrr::reduce(
  list(
    tidy_info,
    pcawg_sbs[, -c(1, 3)] %&gt;% dplyr::rename(sample = `Sample Names`) %&gt;%
      # Drop Possible sequencing artefact associated signatures
      dplyr::select(-SBS43, -c(SBS45:SBS60)),
    pcawg_dbs[, -c(1, 3)] %&gt;% dplyr::rename(sample = `Sample Names`),
    pcawg_id[, -c(1, 3)] %&gt;% dplyr::rename(sample = `Sample Names`),
    cluster_df[, -2] %&gt;% purrr::set_names(c(&quot;sample&quot;, &quot;cluster&quot;))
  ),
  dplyr::left_join,
  by = &quot;sample&quot;
) %&gt;%
  dplyr::filter(!is.na(cluster)) %&gt;%
  mutate(cluster = paste0(&quot;subgroup&quot;, cluster))

colnames(df_merged) &lt;- gsub(&quot;Abs_&quot;, &quot;&quot;, colnames(df_merged))</code></pre>
<div id="enrichment-analysis-for-copy-number-signatures" class="section level4">
<h4>Enrichment analysis for copy number signatures</h4>
<pre class="r"><code>library(sigminer)
enrich_result_cn &lt;- group_enrichment(
  df_merged,
  grp_vars = &quot;cluster&quot;,
  enrich_vars = paste0(&quot;Sig&quot;, 1:11),
  co_method = &quot;wilcox.test&quot;
)</code></pre>
<pre class="r"><code>enrich_result_cn$enrich_var &lt;- factor(enrich_result_cn$enrich_var, paste0(&quot;Sig&quot;, 1:11))

p &lt;- show_group_enrichment(
  enrich_result_cn,
  fill_by_p_value = TRUE,
  cut_p_value = TRUE,
  return_list = T
)
p &lt;- p$cluster + labs(x = NULL, y = NULL)
p + coord_flip()</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-101-1.png" width="1950" /></p>
</div>
<div id="enrichment-analysis-for-sbs-signatures" class="section level4">
<h4>Enrichment analysis for SBS signatures</h4>
<pre class="r"><code>nm_sbs &lt;- colnames(df_merged)[grepl(&quot;SBS&quot;, colnames(df_merged))]

enrich_result_sbs &lt;- group_enrichment(
  df_merged,
  grp_vars = &quot;cluster&quot;,
  enrich_vars = nm_sbs,
  co_method = &quot;wilcox.test&quot;
)</code></pre>
<pre class="r"><code>enrich_result_sbs$enrich_var &lt;- factor(enrich_result_sbs$enrich_var, nm_sbs)

p &lt;- show_group_enrichment(
  enrich_result_sbs,
  fill_by_p_value = TRUE,
  cut_p_value = TRUE,
  return_list = T
)
p &lt;- p$cluster + labs(x = NULL, y = NULL)
p + coord_flip()</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-103-1.png" width="1950" /></p>
</div>
<div id="enrichment-analysis-for-dbs-signatures" class="section level4">
<h4>Enrichment analysis for DBS signatures</h4>
<pre class="r"><code>nm_dbs &lt;- colnames(df_merged)[grepl(&quot;DBS&quot;, colnames(df_merged))]

enrich_result_dbs &lt;- group_enrichment(
  df_merged,
  grp_vars = &quot;cluster&quot;,
  enrich_vars = nm_dbs,
  co_method = &quot;wilcox.test&quot;
)</code></pre>
<pre class="r"><code>enrich_result_dbs$enrich_var &lt;- factor(enrich_result_dbs$enrich_var, nm_dbs)

p &lt;- show_group_enrichment(
  enrich_result_dbs,
  fill_by_p_value = TRUE,
  cut_p_value = TRUE,
  return_list = T
)
p &lt;- p$cluster + labs(x = NULL, y = NULL)
p + coord_flip()</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-105-1.png" width="1950" /></p>
</div>
<div id="enrichment-analysis-for-id-signatures" class="section level4">
<h4>Enrichment analysis for ID signatures</h4>
<pre class="r"><code>nm_id &lt;- colnames(df_merged)[grepl(&quot;^ID[0-9]+&quot;, colnames(df_merged))]

enrich_result_id &lt;- group_enrichment(
  df_merged,
  grp_vars = &quot;cluster&quot;,
  enrich_vars = nm_id,
  co_method = &quot;wilcox.test&quot;
)</code></pre>
<pre class="r"><code>enrich_result_id$enrich_var &lt;- factor(enrich_result_id$enrich_var, nm_id)

p &lt;- show_group_enrichment(
  enrich_result_id,
  fill_by_p_value = TRUE,
  cut_p_value = TRUE,
  return_list = T
)
p &lt;- p$cluster + labs(x = NULL, y = NULL)
p + coord_flip()</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-107-1.png" width="1950" /></p>
</div>
<div id="combined-cosmic-signature-result" class="section level4">
<h4>Combined COSMIC signature result</h4>
<p>Merge all COSMIC signature results.</p>
<pre class="r"><code>enrich_result_cosmic &lt;- purrr::reduce(
  list(enrich_result_sbs, enrich_result_dbs, enrich_result_id),
  rbind
)</code></pre>
<blockquote>
<p>移除带 NA 和 Inf 的 signatures 结果，然后绘制一个汇总图，对 signatures 进行聚类，以确定 signature 排序。</p>
</blockquote>
</div>
<div id="word-cloud-plot-for-cosmic-signature-etiologies" class="section level4">
<h4>Word cloud plot for COSMIC signature etiologies</h4>
<p>There are so many COSMIC signatures above, it is not easy to summarize the data. Considering all COSMIC signatures are labeled and many of them have been assigned to a specific etiology, here we try to use word cloud plot to summarize the result above.</p>
<p>Load etiologies of COSMIC signatures.</p>
<pre class="r"><code>cosmic_ets &lt;- purrr::reduce(
  list(
    sigminer::get_sig_db(&quot;SBS&quot;)$aetiology %&gt;% tibble::rownames_to_column(&quot;sig_name&quot;),
    sigminer::get_sig_db(&quot;DBS&quot;)$aetiology %&gt;% tibble::rownames_to_column(&quot;sig_name&quot;),
    sigminer::get_sig_db(&quot;ID&quot;)$aetiology %&gt;% tibble::rownames_to_column(&quot;sig_name&quot;)
  ),
  rbind
)</code></pre>
<pre class="r"><code>DT::datatable(cosmic_ets)</code></pre>
<div id="htmlwidget-18f151ff8c7d6fa42bf4" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-18f151ff8c7d6fa42bf4">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101"],["SBS1","SBS2","SBS3","SBS4","SBS5","SBS6","SBS7a","SBS7b","SBS7c","SBS7d","SBS8","SBS9","SBS10a","SBS10b","SBS11","SBS12","SBS13","SBS14","SBS15","SBS16","SBS17a","SBS17b","SBS18","SBS19","SBS20","SBS21","SBS22","SBS23","SBS24","SBS25","SBS26","SBS27","SBS28","SBS29","SBS30","SBS31","SBS32","SBS33","SBS34","SBS35","SBS36","SBS37","SBS38","SBS39","SBS40","SBS41","SBS42","SBS43","SBS44","SBS45","SBS46","SBS47","SBS48","SBS49","SBS50","SBS51","SBS52","SBS53","SBS54","SBS55","SBS56","SBS57","SBS58","SBS59","SBS60","SBS84","SBS85","SBS86","SBS87","SBS88","SBS89","SBS90","DBS1","DBS2","DBS3","DBS4","DBS5","DBS6","DBS7","DBS8","DBS9","DBS10","DBS11","ID1","ID2","ID3","ID4","ID5","ID6","ID7","ID8","ID9","ID10","ID11","ID12","ID13","ID14","ID15","ID16","ID17","ID18"],["Spontaneous deamination of 5-methylcytosine (clock-like signature)","Activity of APOBEC family of cytidine deaminases","Defective homologous recombination DNA damage repair","Tobacco smoking","Unknown (clock-like signature)","Defective DNA mismatch repair","Ultraviolet light exposure","Ultraviolet light exposure","Ultraviolet light exposure","Ultraviolet light exposure","Unknown","Polimerase eta somatic hypermutation activity","Polymerase epsilon exonuclease domain mutations","Polymerase epsilon exonuclease domain mutations","Temozolomide treatment","Unknown","Activity of APOBEC family of cytidine deaminases","Concurrent polymerase epsilon mutation and defective DNA mismatch repair","Defective DNA mismatch repair","Unknown","Unknown","Unknown","Damage by reactive oxygen species","Unknown","Concurrent POLD1 mutations and defective DNA mismatch repair","Defective DNA mismatch repair","Aristolochic acid exposure","Unknown","Aflatoxin exposure","Chemotherapy treatment","Defective DNA mismatch repair","Possible sequencing artefact","Unknown","Tobacco chewing","Defective DNA base excision repair due to NTHL1 mutations","Platinum chemotherapy treatment","Azathioprine treatment","Unknown","Unknown","Platinum chemotherapy treatment","Defective DNA base excision repair due to MUTYH mutations","Unknown","Indirect effect of ultraviolet light","Unknown","Unknown","Unknown","Haloalkane exposure","Possible sequencing artefact","Defective DNA mismatch repair","Possible sequencing artefact","Possible sequencing artefact","Possible sequencing artefact","Possible sequencing artefact","Possible sequencing artefact","Possible sequencing artefact","Possible sequencing artefact","Possible sequencing artefact","Possible sequencing artefact","Possible sequencing artefact","Possible sequencing artefact","Possible sequencing artefact","Possible sequencing artefact","Possible sequencing artefact","Possible sequencing artefact","Possible sequencing artefact","Activity of activation-induced cytidine deaminase (AID)","Indirect effects of activation-induced cytidine deaminase (AID)","Unknown chemotherapy treatment","Thiopurine chemotherapy treatment","Colibactin exposure (E.coli bacteria carrying pks pathogenicity island)","Unknown","Duocarmycin exposure","Ultraviolet light exposure","Tobacco smoking and other mutagens","Polymerase epsilon exonuclease domain mutations","Unknown","Platinum chemotherapy treatment","Unknown","Defective DNA mismatch repair","Unknown","Unknown","Defective DNA mismatch repair","Unknown (possibly related to APOBEC mutagenesis)","Slippage during DNA replication of the replicated DNA strand","Slippage during DNA replication of the template DNA strand","Tobacco smoking","Unknown","Unknown","Defective homologous recombination DNA damage repair","Defective DNA mismatch repair","Repair of DNA double strand breaks by NHEJ mechanisms or mutations in topoisomerase TOP2A","Unknown","Unknown","Unknown","Unknown","Ultraviolet light exposure","Unknown","Unknown","Unknown","Mutations in topoisomerase TOP2A","Colibactin exposure (E.coli bacteria carrying pks pathogenicity island)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>sig_name<\/th>\n      <th>aetiology<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script>
<p>Generate a <code>data.frame</code> for plotting. We only keep significantly and positively enriched signatures in each subgroup. We note that descriptions for many signatures are too long, thus we use short names.</p>
<pre class="r"><code>df_ets &lt;- enrich_result_cosmic %&gt;%
  dplyr::filter(p_value &lt; 0.05 &amp; measure_observed &gt; 1) %&gt;%
  dplyr::select(grp1, enrich_var) %&gt;%
  dplyr::left_join(cosmic_ets, by = c(&quot;enrich_var&quot; = &quot;sig_name&quot;)) %&gt;%
  dplyr::mutate(
    aetiology = dplyr::case_when(
      grepl(&quot;APOBEC&quot;, aetiology) ~ &quot;APOBEC&quot;,
      grepl(&quot;Tobacco&quot;, aetiology) ~ &quot;Tobacco&quot;,
      grepl(&quot;clock&quot;, aetiology) ~ &quot;Clock&quot;,
      grepl(&quot;Ultraviolet&quot;, aetiology, ignore.case = TRUE) ~ &quot;UV&quot;,
      grepl(&quot;homologous recombination&quot;, aetiology) ~ &quot;HRD&quot;,
      grepl(&quot;mismatch repair&quot;, aetiology) ~ &quot;dMMR&quot;,
      grepl(&quot;Slippage&quot;, aetiology) ~ &quot;Slippage&quot;,
      grepl(&quot;base excision repair&quot;, aetiology) ~ &quot;dBER&quot;,
      grepl(&quot;NHEJ&quot;, aetiology) ~ &quot;NHEJ repair&quot;,
      grepl(&quot;reactive oxygen&quot;, aetiology) ~ &quot;ROS&quot;,
      grepl(&quot;Polymerase epsilon exonuclease&quot;, aetiology) ~ &quot;Polymerase epsilon mutation&quot;,
      grepl(&quot;eta somatic hypermutation&quot;, aetiology) ~ &quot;Polimerase eta hypermutation&quot;,
      TRUE ~ aetiology
    )
  ) %&gt;%
  dplyr::count(grp1, aetiology) %&gt;%
  dplyr::filter(!aetiology %in% c(&quot;Unknown&quot;, &quot;Possible sequencing artefact&quot;))</code></pre>
<p>Plotting.</p>
<pre class="r"><code>library(ggwordcloud)

set.seed(42)
ggplot(df_ets, aes(label = aetiology, size = n)) +
  geom_text_wordcloud_area(eccentricity = .35, shape = &quot;square&quot;) +
  scale_size_area(max_size = 14) +
  facet_wrap(~grp1) +
  theme_bw()</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-112-1.png" width="3000" /></p>
</div>
</div>
<div id="other-integrated-variables" class="section level3">
<h3>Other integrated variables</h3>
<pre class="r"><code>df_others &lt;- readRDS(&quot;../data/pcawg_tidy_anno_df.rds&quot;) %&gt;%
  tibble::rownames_to_column(&quot;sample&quot;) %&gt;%
  dplyr::left_join(
    cluster_df[, -2] %&gt;% purrr::set_names(c(&quot;sample&quot;, &quot;cluster&quot;)),
    by = &quot;sample&quot;
  ) %&gt;%
  dplyr::mutate(
    Tumor_Stage = dplyr::case_when(
      Tumor_Stage == &quot;I&quot; ~ 1,
      Tumor_Stage == &quot;II&quot; ~ 2,
      Tumor_Stage == &quot;III&quot; ~ 3,
      Tumor_Stage == &quot;IV&quot; ~ 4
    ),
    isFemale = Sex == &quot;female&quot;,
    isWGD = WGD_Status == &quot;wgd&quot;,
    isHRD = HRD_Status == &quot;hrd&quot;,
    cluster = paste0(&quot;subgroup&quot;, cluster)
  ) %&gt;%
  dplyr::select(-c(Sex, WGD_Status, HRD_Status))</code></pre>
<pre class="r"><code>nm_others &lt;- setdiff(colnames(df_others), c(&quot;sample&quot;, &quot;cluster&quot;))

enrich_result_others &lt;- group_enrichment(
  df_others,
  grp_vars = &quot;cluster&quot;,
  enrich_vars = nm_others,
  co_method = &quot;wilcox.test&quot;
)</code></pre>
<pre class="r"><code>enrich_result_others$enrich_var &lt;- factor(enrich_result_others$enrich_var, nm_others)

p &lt;- show_group_enrichment(
  enrich_result_others,
  fill_by_p_value = TRUE,
  cut_p_value = TRUE,
  return_list = T
)
p &lt;- p$cluster + labs(x = NULL, y = NULL)
p + coord_flip()</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-115-1.png" width="1950" /></p>
<blockquote>
<p>The <code>HR</code> here represents <code>Heavily rearrangement</code>, don’t mix it with <code>HRD</code> which means <code>Defective homologous recombination DNA damage repair</code>.</p>
</blockquote>
</div>
<div id="driver-genes" class="section level3">
<h3>Driver genes</h3>
<p>We have already observed that there are different mutational processes across different groups. Here we try to go further study if different driver genes activate in different groups.</p>
<p>We obtained coding driver mutations from <a href="https://xena.hiplot.com.cn/datapages/?dataset=pcawg_whitelist_coding_drivers_v1_sep302016.sp.xena&amp;host=https%3A%2F%2Fxena-pcawg.hiplot.com.cn">UCSC Xena database</a>. For known driver genes, we focus on driver gene list obtained from <span class="citation"><a href="#ref-martinez-jimenezCompendiumMutationalCancer2020" role="doc-biblioref">Martínez-Jiménez et al.</a> (<a href="#ref-martinez-jimenezCompendiumMutationalCancer2020" role="doc-biblioref">2020</a>)</span>.</p>
<pre class="r"><code>load(&quot;../data/pcawg_driver_info.RData&quot;)
df_mut &lt;- dplyr::left_join(df_others[, c(&quot;sample&quot;, &quot;cluster&quot;)],
  pcawg_driver_number,
  by = &quot;sample&quot;
)
colnames(df_mut)[3] &lt;- &quot;Driver_variants&quot;
df_mut$Driver_variants &lt;- ifelse(is.na(df_mut$Driver_variants), 0, df_mut$Driver_variants)

driver_gene_num &lt;- pcawg_driver_gene %&gt;%
  dplyr::group_by(sample, ROLE) %&gt;%
  dplyr::summarise(driver_genes = n(), .groups = &quot;drop&quot;) %&gt;%
  tidyr::pivot_wider(names_from = &quot;ROLE&quot;, values_from = &quot;driver_genes&quot;, values_fill = 0)

df_mut &lt;- dplyr::left_join(df_mut, driver_gene_num, by = &quot;sample&quot;)
colnames(df_mut)[4:5] &lt;- c(&quot;Act_genes&quot;, &quot;LoF_genes&quot;)
df_mut$Act_genes &lt;- ifelse(is.na(df_mut$Act_genes), 0, df_mut$Act_genes)
df_mut$LoF_genes &lt;- ifelse(is.na(df_mut$LoF_genes), 0, df_mut$LoF_genes)</code></pre>
<p>Also merge into info of each driver gene.</p>
<pre class="r"><code>df_mut2 &lt;- df_mut %&gt;%
  dplyr::left_join(
    pcawg_driver_gene %&gt;%
      tidyr::pivot_wider(names_from = &quot;gene&quot;, values_from = &quot;ROLE&quot;) %&gt;%
      dplyr::mutate_at(vars(-sample), ~ !is.na(.)),
    by = &quot;sample&quot;
  )

## Filter out driver genes with &lt; 10 mutation
gene_sum &lt;- sort(colSums(df_mut2[, -c(1:5)], na.rm = TRUE), decreasing = TRUE)


# 基因按突变数量排序

df_mut2 &lt;- df_mut2 %&gt;% dplyr::select_at(c(colnames(df_mut2)[1:5], names(gene_sum[gene_sum &gt;= 10])))</code></pre>
<p>Save the data.</p>
<pre class="r"><code>saveRDS(df_mut2, file = &quot;../data/pcawg_mut_df.rds&quot;)</code></pre>
<pre class="r"><code>nm_mut &lt;- setdiff(colnames(df_mut2), c(&quot;sample&quot;, &quot;cluster&quot;))

enrich_result_mut &lt;- group_enrichment(
  df_mut2,
  grp_vars = &quot;cluster&quot;,
  enrich_vars = nm_mut,
  co_method = &quot;wilcox.test&quot;
)</code></pre>
<p>Filter out genes insignificant in all subgroups.</p>
<pre class="r"><code>keep_df &lt;- enrich_result_mut[, .(count = sum(p_value &lt; 0.05)), by = enrich_var]
drop_vars &lt;- keep_df[count == 0]$enrich_var

enrich_result_mut2 &lt;- enrich_result_mut[!enrich_var %in% drop_vars]</code></pre>
<p>Plotting.</p>
<pre class="r"><code>enrich_result_mut2$enrich_var &lt;- factor(enrich_result_mut2$enrich_var, setdiff(nm_mut, drop_vars))

keep_vars &lt;- setdiff(nm_mut, drop_vars)[-c(1:3)]
labels &lt;- c(nm_mut[1:3], paste0(keep_vars, &quot; (&quot;, gene_sum[keep_vars], &quot;)&quot;))

p &lt;- show_group_enrichment(
  enrich_result_mut2,
  fill_by_p_value = TRUE,
  cut_p_value = TRUE,
  return_list = T
)
p &lt;- p$cluster + labs(x = NULL, y = NULL)
p + coord_flip() + scale_x_discrete(labels = labels)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-122-1.png" width="1950" /></p>
<pre class="r"><code>rm(list = ls())</code></pre>
</div>
</div>
</div>
<div id="part-5-inference-of-copy-number-signature-etiologies" class="section level1">
<h1>PART 5: Inference of copy number signature etiologies</h1>
<p>In this part, we would like to explore CN signatures in more details with association analysis and summarize what we get to infer their etiologies.</p>
<p>Let’s merge load all data we got and combine them.</p>
<pre class="r"><code>library(tidyverse)
library(sigminer)

tidy_info &lt;- readRDS(&quot;../data/pcawg_sample_tidy_info.rds&quot;)
tidy_anno &lt;- readRDS(&quot;../data/pcawg_tidy_anno_df.rds&quot;)
tidy_mut &lt;- readRDS(&quot;../data/pcawg_mut_df.rds&quot;)
pcawg_sbs &lt;- read_csv(&quot;../data/PCAWG/PCAWG_sigProfiler_SBS_signatures_in_samples.csv&quot;)
pcawg_dbs &lt;- read_csv(&quot;../data/PCAWG/PCAWG_sigProfiler_DBS_signatures_in_samples.csv&quot;)
pcawg_id &lt;- read_csv(&quot;../data/PCAWG/PCAWG_SigProfiler_ID_signatures_in_samples.csv&quot;)</code></pre>
<p>Merge data.</p>
<pre class="r"><code>df_merged &lt;- purrr::reduce(
  list(
    tidy_info %&gt;%
      dplyr::filter(keep) %&gt;%
      dplyr::select(sample, starts_with(&quot;Abs&quot;), starts_with(&quot;Rel&quot;)),
    tidy_anno %&gt;%
      tibble::rownames_to_column(&quot;sample&quot;) %&gt;%
      dplyr::mutate(
        Tumor_Stage = dplyr::case_when(
          Tumor_Stage == &quot;I&quot; ~ 1,
          Tumor_Stage == &quot;II&quot; ~ 2,
          Tumor_Stage == &quot;III&quot; ~ 3,
          Tumor_Stage == &quot;IV&quot; ~ 4
        ),
        isFemale = Sex == &quot;female&quot;,
        isWGD = WGD_Status == &quot;wgd&quot;,
        isHRD = HRD_Status == &quot;hrd&quot;
      ) %&gt;%
      dplyr::select(-c(Sex, WGD_Status, HRD_Status)),
    pcawg_sbs[, -c(1, 3)] %&gt;% dplyr::rename(sample = `Sample Names`) %&gt;%
      # Drop Possible sequencing artefact associated signatures
      dplyr::select(-SBS43, -c(SBS45:SBS60)),
    pcawg_dbs[, -c(1, 3)] %&gt;% dplyr::rename(sample = `Sample Names`),
    pcawg_id[, -c(1, 3)] %&gt;% dplyr::rename(sample = `Sample Names`),
    tidy_mut %&gt;% dplyr::select(-cluster)
  ),
  dplyr::left_join,
  by = &quot;sample&quot;
) %&gt;%
  dplyr::mutate_if(
    is.logical,
    ~ ifelse(is.na(.), FALSE, .)
  )

colnames(df_merged) &lt;- gsub(&quot;Abs_&quot;, &quot;&quot;, colnames(df_merged))

saveRDS(df_merged, file = &quot;../data/pcawg_merged_df.rds&quot;)</code></pre>
<div id="association-analysis-for-copy-number-signatures" class="section level2">
<h2>Association analysis for copy number signatures</h2>
<div id="copy-number-signature-activity-self-association" class="section level3">
<h3>Copy number signature activity self-association</h3>
<p>For absolute signature activity.</p>
<pre class="r"><code>show_cor(dplyr::select(df_merged, starts_with(&quot;Sig&quot;)))</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-129-1.png" width="2400" /></p>
<p>For relative signature activity.</p>
<pre class="r"><code>show_cor(dplyr::select(df_merged, starts_with(&quot;Rel_Sig&quot;)) %&gt;%
  dplyr::rename_all(~ sub(&quot;Rel_&quot;, &quot;&quot;, .)))</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-130-1.png" width="2400" /> We can find that most the signatures are correlated. Therefore, it will add difficulty in understanding etiologies by association analysis.</p>
</div>
<div id="copy-number-signatures-and-cosmic-signatures" class="section level3">
<h3>Copy number signatures and COSMIC signatures</h3>
<pre class="r"><code>p_cosmic &lt;- show_cor(
  data = df_merged,
  x_vars = colnames(df_merged)[grepl(&quot;^Sig&quot;, colnames(df_merged))],
  y_vars = colnames(df_merged)[grepl(&quot;^SBS|^DBS|^ID[0-9]+&quot;, colnames(df_merged))],
  hc_order = FALSE,
  test = FALSE,
  lab_size = 2
)

p_cosmic</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-131-1.png" width="1800" /></p>
</div>
<div id="signature-clustering" class="section level3">
<h3>Signature clustering</h3>
<p>We can take a try to cluster signatures based on their activity association.</p>
<p>First get the correlation data.</p>
<pre class="r"><code>p_all &lt;- show_cor(
  data = df_merged,
  x_vars = colnames(df_merged)[grepl(&quot;^Sig|^SBS|^DBS|^ID[0-9]+&quot;, colnames(df_merged))],
  y_vars = colnames(df_merged)[grepl(&quot;^Sig|^SBS|^DBS|^ID[0-9]+&quot;, colnames(df_merged))],
  hc_order = FALSE,
  test = FALSE,
  lab_size = 3
)</code></pre>
<p>Get the correlation matrix and convert it into distance matrix.</p>
<pre class="r"><code>p_all_cor &lt;- p_all$cor$cor_mat
all_dist &lt;- 1 - p_all_cor
# Set the NA value to 2
all_dist[is.na(all_dist)] &lt;- 2</code></pre>
<div id="only-include-copy-number-signatures" class="section level4">
<h4>Only include copy number signatures</h4>
<pre class="r"><code>dist_cn &lt;- as.dist(all_dist[1:11, 1:11])
hc_cn &lt;- hclust(dist_cn)
plot(hc_cn)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-134-1.png" width="2400" /></p>
<p>Roughly there is 3 clusters.</p>
<pre class="r"><code>factoextra::fviz_dend(hc_cn, k = 3, rect = T)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-135-1.png" width="2400" /></p>
</div>
<div id="all-signatures" class="section level4">
<h4>All signatures</h4>
<pre class="r"><code>dist_all &lt;- as.dist(all_dist)
hc_all &lt;- hclust(dist_all)
plot(hc_all)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-136-1.png" width="3600" /> Here we manually set 10 clusters for better visualization.</p>
<pre class="r"><code>factoextra::fviz_dend(
  hc_all,
  k = 10,
  cex = 0.6, rect = F,
  type = &quot;circular&quot;
) + theme_void() +
  theme(
    panel.border = element_blank()
  )</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-137-1.png" width="2400" /></p>
<pre class="r"><code># theme(
#   panel.background = element_rect(fill = &quot;transparent&quot;, colour = NA),
#   plot.background = element_rect(fill = &quot;transparent&quot;, colour = NA),
#   panel.grid = element_blank(),
#   panel.border = element_blank(),
#   plot.margin = unit(c(0, 0, 0, 0), &quot;null&quot;),
#   panel.margin = unit(c(0, 0, 0, 0), &quot;null&quot;),
#   axis.ticks = element_blank(),
#   axis.text = element_blank(),
#   axis.title = element_blank(),
#   axis.line = element_blank(),
#   legend.position = &quot;none&quot;,
#   axis.ticks.length = unit(0, &quot;null&quot;),
#   axis.ticks.margin = unit(0, &quot;null&quot;),
#   legend.margin = unit(0, &quot;null&quot;)
# )</code></pre>
</div>
</div>
<div id="association-between-cns-and-genotypephenotype-features" class="section level3">
<h3>Association between CNS and genotype/phenotype features</h3>
<div id="association-between-cns-and-continuous-variables" class="section level4">
<h4>Association between CNS and continuous variables</h4>
<pre class="r"><code>p_cn_others &lt;- show_cor(
  data = df_merged,
  x_vars = colnames(df_merged)[2:12],
  y_vars = colnames(df_merged)[c(24:38, 119:121)],
  lab_size = 3
)
p_cn_others</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-138-1.png" width="2700" /></p>
</div>
<div id="association-between-cns-and-discrete-variables" class="section level4">
<h4>Association between CNS and discrete variables</h4>
<p>Here we directly compare the CNS activity between different discrete variable groups.</p>
<p>Here 3 variables including <code>WGD_Status</code>, <code>HRD_Status</code> and <code>Tumor_Stage</code> are included.</p>
<pre class="r"><code>df_ds &lt;- tidy_info %&gt;%
  dplyr::filter(keep) %&gt;%
  dplyr::select(sample, starts_with(&quot;Abs&quot;)) %&gt;%
  dplyr::mutate_if(is.numeric, ~ log2(. + 1)) %&gt;%
  purrr::set_names(c(&quot;sample&quot;, paste0(&quot;Sig&quot;, 1:11))) %&gt;%
  dplyr::left_join(
    tidy_anno %&gt;%
      tibble::rownames_to_column(&quot;sample&quot;) %&gt;%
      dplyr::select(sample, WGD_Status, HRD_Status, Tumor_Stage),
    by = &quot;sample&quot;
  ) %&gt;%
  dplyr::mutate(Tumor_Stage = ifelse(Tumor_Stage == &quot;Unknown&quot;, NA, Tumor_Stage)) %&gt;%
  tidyr::pivot_longer(
    starts_with(&quot;Sig&quot;),
    names_to = &quot;sig_name&quot;,
    values_to = &quot;activity&quot;
  )</code></pre>
<pre class="r"><code>library(ggpubr)</code></pre>
<p>For WGD.</p>
<pre class="r"><code>ggboxplot(
  df_ds,
  x = &quot;sig_name&quot;, y = &quot;activity&quot;,
  color = &quot;WGD_Status&quot;,
  palette = &quot;jco&quot;, width = .3,
  xlab = FALSE, ylab = &quot;log2(Activity)&quot;,
  outlier.size = 0.05
) +
  stat_compare_means(aes(group = WGD_Status), label = &quot;p.signif&quot;)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-141-1.png" width="2400" /> For HRD.</p>
<pre class="r"><code>ggboxplot(
  df_ds %&gt;%
    dplyr::mutate(HRD_Status = factor(HRD_Status, c(&quot;no_hrd&quot;, &quot;hrd&quot;))),
  x = &quot;sig_name&quot;, y = &quot;activity&quot;,
  color = &quot;HRD_Status&quot;,
  palette = &quot;jco&quot;, width = .3,
  xlab = FALSE, ylab = &quot;log2(Activity)&quot;,
  outlier.size = 0.05
) +
  stat_compare_means(aes(group = HRD_Status), label = &quot;p.signif&quot;)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-142-1.png" width="2400" /></p>
<p>For tumor stage.</p>
<pre class="r"><code>ggboxplot(
  df_ds %&gt;% dplyr::filter(!is.na(Tumor_Stage)),
  x = &quot;sig_name&quot;, y = &quot;activity&quot;,
  color = &quot;Tumor_Stage&quot;,
  palette = &quot;jco&quot;, width = .3,
  xlab = FALSE, ylab = &quot;log2(Activity)&quot;,
  outlier.size = 0.05
) +
  stat_compare_means(aes(group = Tumor_Stage),
    label = &quot;p.signif&quot;,
    method = &quot;anova&quot;
  )</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-143-1.png" width="2400" /></p>
</div>
<div id="association-between-cns-and-driver-gene-mutation" class="section level4">
<h4>Association between CNS and driver gene mutation</h4>
<pre class="r"><code>asso_res &lt;- get_sig_feature_association(
  df_merged,
  cols_to_sigs = paste0(&quot;Sig&quot;, 1:11),
  cols_to_features = colnames(df_merged)[122:205],
  method_ca = wilcox.test,
  min_n = 0
)

asso_res_tidy &lt;- get_tidy_association(asso_res)</code></pre>
<p>Plotting.</p>
<pre class="r"><code>show_sig_feature_corrplot(
  asso_res_tidy,
  sig_orders = paste0(&quot;Sig&quot;, 1:11),
  p_val = 0.05,
  breaks_count = NA,
  ylab = NULL,
  xlab = NULL
) + ggpubr::rotate_x_text(0, hjust = 0.5) +
  labs(color = &quot;Difference\nin means of\nactivity&quot;)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-145-1.png" width="2700" /></p>
</div>
</div>
<div id="association-between-cns-and-patients-prognosis" class="section level3">
<h3>Association between CNS and patients’ prognosis</h3>
<p>Get tidy data fro survival analysis.</p>
<pre class="r"><code>library(ezcox)

cluster_df &lt;- readRDS(&quot;../data/pcawg_clusters.rds&quot;) %&gt;%
  as.data.frame() %&gt;%
  tibble::rownames_to_column(&quot;sample&quot;)

df_surv &lt;- tidy_info %&gt;%
  dplyr::filter(keep) %&gt;%
  dplyr::select(
    sample, cancer_type, donor_survival_time, donor_vital_status,
    dplyr::starts_with(&quot;Abs_&quot;)
  ) %&gt;%
  dplyr::left_join(
    cluster_df,
    by = &quot;sample&quot;
  ) %&gt;%
  na.omit() %&gt;%
  dplyr::rename(
    time = donor_survival_time,
    status = donor_vital_status,
    cluster = `k=6`
  ) %&gt;%
  dplyr::mutate(
    status = ifelse(status == &quot;deceased&quot;, 1, 0)
  ) %&gt;%
  dplyr::rename_at(vars(starts_with(&quot;Abs_&quot;)), ~ sub(&quot;Abs_&quot;, &quot;&quot;, .)) %&gt;%
  dplyr::mutate_at(vars(starts_with(&quot;Sig&quot;)), ~ . / 10)</code></pre>
<p>We divide all signature activity by 10 to explore the HR change per 10 records contributed by a signature.</p>
<div id="unicox-analysis-for-cns" class="section level4">
<h4>Unicox analysis for CNS</h4>
<p>Here we focus on the association between each CNS and patients’ overall survival time.</p>
<pre class="r"><code>p1 &lt;- show_forest(df_surv,
  covariates = paste0(&quot;Sig&quot;, 1:11),
  merge_models = TRUE, add_caption = FALSE, point_size = 2
)
p1</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-147-1.png" width="2400" /></p>
</div>
<div id="multivariate-cox-analysis-for-cns" class="section level4">
<h4>Multivariate cox analysis for CNS</h4>
<p>We also build a multivariate cox model to explore if all CNSs are included, which is more important?</p>
<pre class="r"><code>p2 &lt;- show_forest(df_surv,
  covariates = &quot;Sig1&quot;, controls = paste0(&quot;Sig&quot;, 2:11),
  merge_models = TRUE, add_caption = FALSE, point_size = 2
)
p2</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-148-1.png" width="2400" /></p>
</div>
</div>
</div>
<div id="signature-similarity-comparison-between-different-types-of-signatures" class="section level2">
<h2>Signature similarity comparison between different types of signatures</h2>
<p>The similarity between signatures is a key parameter, it determines how hard it is to distinguish between signatures and the precision to assign the contribution to a signature. If you have read the paper <span class="citation"><a href="#ref-mauraPracticalGuideMutational2019" role="doc-biblioref">Maura et al.</a> (<a href="#ref-mauraPracticalGuideMutational2019" role="doc-biblioref">2019</a>)</span>, you may know that 2 of 3 main issues proposed by the authors are actually because of high similarity between mutational signatures:</p>
<ul>
<li><strong>the ambiguous signature assignment</strong> that occurs when different combinations of signatures can explain equally well the same mutational catalog.</li>
<li><strong>bleeding of signatures</strong> that signatures present in only part of the set are also erroneously assigned to the entire set.</li>
</ul>
<p>In this section, we explore the similarity of CNSs discovered in this study and known COSMIC signatures.</p>
<p>Load data.</p>
<pre class="r"><code>CNS &lt;- readRDS(&quot;../data/pcawg_cn_sigs_CN176_signature.rds&quot;)
SBSv2 &lt;- get_sig_db()
SBS &lt;- get_sig_db(&quot;SBS&quot;)
DBS &lt;- get_sig_db(&quot;DBS&quot;)
ID &lt;- get_sig_db(&quot;ID&quot;)</code></pre>
<p>Get signature similarity in each signature type.</p>
<pre class="r"><code>sim_CNS &lt;- get_sig_similarity(CNS, CNS)$similarity
sim_SBSv2 &lt;- get_sig_similarity(SBSv2$db, SBSv2$db)$similarity
sim_SBS &lt;- get_sig_similarity(SBS$db, SBS$db)$similarity
sim_DBS &lt;- get_sig_similarity(DBS$db, DBS$db)$similarity
sim_ID &lt;- get_sig_similarity(ID$db, ID$db)$similarity</code></pre>
<p>Only use upper matrix to get similarity between one signature and non-self signatures.</p>
<pre class="r"><code>a &lt;- sim_CNS[upper.tri(sim_CNS)]
b &lt;- sim_SBSv2[upper.tri(sim_SBSv2)]
c &lt;- sim_SBS[upper.tri(sim_SBS)]
d &lt;- sim_DBS[upper.tri(sim_DBS)]
e &lt;- sim_ID[upper.tri(sim_ID)]


df_sim &lt;- dplyr::tibble(
  sig_type = rep(
    c(&quot;CNS&quot;, &quot;SBSv2&quot;, &quot;SBS&quot;, &quot;DBS&quot;, &quot;ID&quot;),
    c(length(a), length(b), length(c), length(d), length(e))
  ),
  similarity = c(a, b, c, d, e)
)</code></pre>
<p>See header rows:</p>
<pre class="r"><code>head(df_sim)</code></pre>
<pre><code># A tibble: 6 x 2
  sig_type similarity
  &lt;chr&gt;         &lt;dbl&gt;
1 CNS           0.455
2 CNS           0.015
3 CNS           0    
4 CNS           0.131
5 CNS           0.001
6 CNS           0.059</code></pre>
<p>Let’s visualize it and use ‘CNS’ as reference group to run statistical test (‘wilcox.test’ method).</p>
<pre class="r"><code>library(ggpubr)

ggboxplot(df_sim,
  x = &quot;sig_type&quot;, y = &quot;similarity&quot;,
  color = &quot;sig_type&quot;,
  xlab = FALSE,
  ylab = &quot;Similarity between signatures&quot;,
  legend = &quot;none&quot;,
  palette = &quot;jco&quot;,
  width = 0.3,
  outlier.size = 0.05
) +
  # stat_compare_means(method = &quot;anova&quot;) +
  stat_compare_means(
    label = &quot;p.signif&quot;, method = &quot;wilcox.test&quot;,
    ref.group = &quot;CNS&quot;
  )</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-153-1.png" width="1200" /> We can see that signature type CNS has lowest similarity level and CNS signatures are not likely similar to others. CNS has much lower similarity level than all other signature types except DBS.</p>
</div>
<div id="data-summary-and-inference-of-signature-etiologies" class="section level2">
<h2>Data summary and inference of signature etiologies</h2>
<p>This is a summary section, we utilize all data we got from analyses in this study and infer how should we explore each copy number signature.</p>
<p>We will list all CNSs, and their features, associations and our inferences here.</p>
<blockquote>
<p>TODO</p>
</blockquote>
</div>
</div>
<div id="supplementary-analyses" class="section level1">
<h1>Supplementary Analyses</h1>
<p>In this part, we will add some supplementary analyses.</p>
<div id="signature-association-network-construction-and-visualization" class="section level2">
<h2>Signature association network construction and visualization</h2>
<blockquote>
<p>Try to learn from NC paper</p>
</blockquote>
</div>
<div id="tcga-cns-analysis" class="section level2">
<h2>TCGA CNS analysis</h2>
<blockquote>
<p>Think and check in future</p>
</blockquote>
<div id="copy-number-signatures" class="section level3">
<h3>Copy number signatures</h3>
</div>
<div id="cancer-subtyping" class="section level3">
<h3>Cancer subtyping</h3>
</div>
<div id="prognosis" class="section level3">
<h3>Prognosis</h3>
</div>
</div>
<div id="exploration-of-differnt-signature-extraction-methods" class="section level2">
<h2>Exploration of differnt signature extraction methods</h2>
<blockquote>
<p>Similarity for same number of signatures</p>
</blockquote>
</div>
</div>
<div id="r-session" class="section level1">
<h1>R Session:</h1>
<p><em>Many thanks to authors and contributors of all packages used in this project</em>.</p>
<pre class="r"><code>devtools::session_info()</code></pre>
<pre><code>─ Session info ───────────────────────────────────────────────────────────────
 setting  value                       
 version  R version 4.0.2 (2020-06-22)
 os       macOS High Sierra 10.13.6   
 system   x86_64, darwin17.0          
 ui       X11                         
 language EN                          
 collate  zh_CN.UTF-8                 
 ctype    zh_CN.UTF-8                 
 tz       Asia/Shanghai               
 date     2020-12-31                  

─ Packages ───────────────────────────────────────────────────────────────────
 package        * version    date       lib
 abind            1.4-5      2016-07-21 [1]
 annotate         1.66.0     2020-04-28 [1]
 AnnotationDbi    1.50.3     2020-07-25 [1]
 assertthat       0.2.1      2019-03-21 [1]
 backports        1.2.1      2020-12-09 [1]
 bigmemory      * 4.5.36     2019-12-23 [1]
 bigmemory.sri    0.1.3      2014-08-18 [1]
 Biobase        * 2.48.0     2020-04-27 [1]
 BiocGenerics   * 0.34.0     2020-04-27 [1]
 BiocManager      1.30.10    2019-11-16 [1]
 bit              4.0.4      2020-08-04 [1]
 bit64            4.0.5      2020-08-30 [1]
 bitops           1.0-6      2013-08-17 [1]
 blob             1.2.1      2020-01-20 [1]
 bookdown         0.21       2020-10-13 [1]
 brew             1.0-6      2011-04-13 [1]
 broom            0.7.3      2020-12-16 [1]
 Cairo            1.5-12.2   2020-07-07 [1]
 callr            3.5.1      2020-10-13 [1]
 car              3.0-10     2020-09-29 [1]
 carData          3.0-4      2020-05-22 [1]
 cellranger       1.1.0      2016-07-27 [1]
 circlize         0.4.11     2020-10-31 [1]
 cli              2.2.0      2020-11-20 [1]
 clue             0.3-58     2020-12-03 [1]
 cluster          2.1.0      2019-06-19 [2]
 codetools        0.2-18     2020-11-04 [2]
 cola           * 1.9.0.1012 2020-12-22 [1]
 colorspace       2.0-0      2020-11-11 [1]
 ComplexHeatmap * 2.7.2.1000 2020-12-22 [1]
 cowplot        * 1.1.0      2020-09-08 [1]
 crayon           1.3.4      2017-09-16 [1]
 crosstalk        1.1.0.1    2020-03-13 [1]
 curl             4.3        2019-12-02 [1]
 data.table       1.13.4     2020-12-08 [1]
 DBI              1.1.0      2019-12-15 [1]
 dbplyr           2.0.0      2020-11-03 [1]
 dendextend       1.14.0     2020-08-26 [1]
 desc             1.2.0      2018-05-01 [1]
 devtools         2.3.2      2020-09-18 [1]
 digest           0.6.27     2020-10-24 [1]
 doParallel       1.0.16     2020-10-16 [1]
 dplyr          * 1.0.2      2020-08-18 [1]
 DT               0.16       2020-10-13 [1]
 ellipsis         0.3.1      2020-05-15 [1]
 eulerr           6.1.0      2020-03-09 [1]
 evaluate         0.14       2019-05-28 [1]
 ezcox          * 0.8.0      2020-09-25 [1]
 factoextra       1.0.7      2020-04-01 [1]
 fansi            0.4.1      2020-01-08 [1]
 farver           2.0.3      2020-01-16 [1]
 forcats        * 0.5.0      2020-03-01 [1]
 foreach          1.5.1      2020-10-15 [1]
 foreign          0.8-80     2020-05-24 [2]
 forestmodel      0.6.2      2020-07-19 [1]
 fs               1.5.0      2020-07-31 [1]
 furrr            0.2.1      2020-10-21 [1]
 future           1.21.0     2020-12-10 [1]
 genefilter       1.70.0     2020-04-27 [1]
 generics         0.1.0      2020-10-31 [1]
 GetoptLong       1.0.5      2020-12-15 [1]
 ggcorrplot       0.1.3      2019-05-19 [1]
 ggplot2        * 3.3.2      2020-06-19 [1]
 ggplotify        0.0.5      2020-03-12 [1]
 ggpubr         * 0.4.0      2020-06-27 [1]
 ggrepel          0.9.0      2020-12-16 [1]
 ggsci            2.9        2018-05-14 [1]
 ggsignif         0.6.0      2019-08-08 [1]
 ggwordcloud    * 0.5.0      2019-06-02 [1]
 GlobalOptions    0.1.2      2020-06-10 [1]
 globals          0.14.0     2020-11-22 [1]
 glue             1.4.2      2020-08-27 [1]
 gridBase         0.4-7      2014-02-24 [1]
 gridExtra        2.3        2017-09-09 [1]
 gridGraphics     0.5-1      2020-12-13 [1]
 gtable           0.3.0      2019-03-25 [1]
 haven            2.3.1      2020-06-01 [1]
 hms              0.5.3      2020-01-08 [1]
 htmltools        0.5.0      2020-06-16 [1]
 htmlwidgets      1.5.3      2020-12-10 [1]
 httr             1.4.2      2020-07-20 [1]
 impute           1.62.0     2020-04-27 [1]
 IRanges          2.22.2     2020-05-21 [1]
 iterators        1.0.13     2020-10-15 [1]
 jsonlite         1.7.2      2020-12-09 [1]
 km.ci            0.5-2      2009-08-30 [1]
 KMsurv           0.1-5      2012-12-03 [1]
 knitr          * 1.30       2020-09-22 [1]
 labeling         0.4.2      2020-10-20 [1]
 languageserver   0.3.9      2020-12-18 [1]
 lattice          0.20-41    2020-04-02 [2]
 lifecycle        0.2.0      2020-03-06 [1]
 listenv          0.8.0      2019-12-05 [1]
 lubridate        1.7.9.2    2020-11-13 [1]
 magick           2.5.2      2020-11-10 [1]
 magrittr         2.0.1      2020-11-17 [1]
 markdown         1.1        2019-08-07 [1]
 Matrix           1.2-18     2019-11-27 [2]
 matrixStats      0.57.0     2020-09-25 [1]
 mclust           5.4.7      2020-11-20 [1]
 memoise          1.1.0      2017-04-21 [1]
 microbenchmark   1.4-7      2019-09-24 [1]
 modelr           0.1.8      2020-05-19 [1]
 munsell          0.5.0      2018-06-12 [1]
 NMF              0.23.0     2020-08-01 [1]
 openxlsx         4.2.3      2020-10-27 [1]
 pacman         * 0.5.1      2019-03-11 [1]
 parallelly       1.22.0     2020-12-13 [1]
 patchwork        1.1.1      2020-12-17 [1]
 pheatmap         1.0.12     2019-01-04 [1]
 pillar           1.4.7      2020-11-20 [1]
 pkgbuild         1.2.0      2020-12-15 [1]
 pkgconfig        2.0.3      2019-09-22 [1]
 pkgload          1.1.0      2020-05-29 [1]
 pkgmaker         0.32.2     2020-10-20 [1]
 plyr             1.8.6      2020-03-03 [1]
 png              0.1-7      2013-12-03 [1]
 prettyunits      1.1.1      2020-01-24 [1]
 processx         3.4.5      2020-11-30 [1]
 ps               1.5.0      2020-12-05 [1]
 purrr          * 0.3.4      2020-04-17 [1]
 R.cache          0.14.0     2019-12-06 [1]
 R.methodsS3      1.8.1      2020-08-26 [1]
 R.oo             1.24.0     2020-08-26 [1]
 R.utils          2.10.1     2020-08-26 [1]
 R6               2.5.0      2020-10-28 [1]
 RColorBrewer     1.1-2      2014-12-07 [1]
 Rcpp             1.0.5      2020-07-06 [1]
 RCurl            1.98-1.2   2020-04-18 [1]
 readr          * 1.4.0      2020-10-05 [1]
 readxl           1.3.1      2019-03-13 [1]
 registry         0.5-1      2019-03-05 [1]
 rematch2         2.1.2      2020-05-01 [1]
 remotes          2.2.0      2020-07-21 [1]
 reprex           0.3.0      2019-05-16 [1]
 reshape2         1.4.4      2020-04-09 [1]
 rio              0.5.16     2018-11-26 [1]
 rjson            0.2.20     2018-06-08 [1]
 rlang            0.4.9      2020-11-26 [1]
 rmarkdown        2.6        2020-12-14 [1]
 rmdformats     * 1.0.0      2020-11-23 [1]
 rngtools         1.5        2020-01-23 [1]
 roxygen2         7.1.1      2020-06-27 [1]
 roxytest       * 0.0.1      2020-06-03 [1]
 rprojroot        2.0.2      2020-11-15 [1]
 RSQLite          2.2.1      2020-09-30 [1]
 rstatix          0.6.0      2020-06-18 [1]
 rstudioapi       0.13       2020-11-12 [1]
 rvcheck          0.1.8      2020-03-01 [1]
 rvest            0.3.6      2020-07-25 [1]
 S4Vectors        0.26.1     2020-05-16 [1]
 scales           1.1.1      2020-05-11 [1]
 sessioninfo      1.1.1      2018-11-05 [1]
 shape            1.4.5      2020-09-13 [1]
 sigminer       * 1.1.0.99   2020-12-28 [1]
 skmeans          0.2-13     2020-12-03 [1]
 slam             0.1-48     2020-12-03 [1]
 stringi          1.5.3      2020-09-09 [1]
 stringr        * 1.4.0      2019-02-10 [1]
 styler           1.3.2      2020-02-23 [1]
 survival       * 3.2-7      2020-09-28 [2]
 survminer      * 0.4.8      2020-07-25 [1]
 survMisc         0.5.5      2018-07-05 [1]
 synchronicity  * 1.3.5      2018-08-21 [1]
 testthat         3.0.1      2020-12-17 [1]
 tibble         * 3.0.4      2020-10-12 [1]
 tidyr          * 1.1.2      2020-08-27 [1]
 tidyselect       1.1.0      2020-05-11 [1]
 tidyverse      * 1.3.0      2019-11-21 [1]
 usethis          2.0.0      2020-12-10 [1]
 utf8             1.1.4      2018-05-24 [1]
 uuid             0.1-4      2020-02-26 [1]
 vctrs            0.3.6      2020-12-17 [1]
 viridis          0.5.1      2018-03-29 [1]
 viridisLite      0.3.0      2018-02-01 [1]
 withr            2.3.0      2020-09-22 [1]
 xfun             0.19       2020-10-30 [1]
 XML              3.99-0.5   2020-07-23 [1]
 xml2             1.3.2      2020-04-23 [1]
 xtable           1.8-4      2019-04-21 [1]
 yaml             2.2.1      2020-02-01 [1]
 zip              2.1.1      2020-08-27 [1]
 zoo              1.8-8      2020-05-02 [1]
 source                                  
 CRAN (R 4.0.0)                          
 Bioconductor                            
 Bioconductor                            
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.0)                          
 Bioconductor                            
 Bioconductor                            
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 Github (jokergoo/cola@a59f2aa)          
 CRAN (R 4.0.2)                          
 Github (jokergoo/ComplexHeatmap@46b5212)
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.0)                          
 local                                   
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 Bioconductor                            
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 Bioconductor                            
 Bioconductor                            
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.2)                          
 Bioconductor                            
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.2)                          
 local                                   
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.0)                          
 CRAN (R 4.0.2)                          
 CRAN (R 4.0.0)                          

[1] /Users/wsx/R_library
[2] /Library/Frameworks/R.framework/Versions/4.0/Resources/library</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="unnumbered">References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-degasperiPracticalFrameworkOnline2020" class="csl-entry">
Degasperi, Andrea, Tauanne Dias Amarante, Jan Czarnecki, Scott Shooter, Xueqing Zou, Dominik Glodzik, Sandro Morganella, et al. 2020. <span>“A Practical Framework and Online Tool for Mutational Signature Analyses Show Intertissue Variation and Driver Dependencies.”</span> <em>Nature Cancer</em> 1 (2): 249–63. <a href="https://doi.org/10.1038/s43018-020-0027-5">https://doi.org/10.1038/s43018-020-0027-5</a>.
</div>
<div id="ref-huangDetectingPresenceMutational2018" class="csl-entry">
Huang, Xiaoqing, Damian Wojtowicz, and Teresa M Przytycka. 2018. <span>“Detecting Presence of Mutational Signatures in Cancer with Confidence.”</span> Edited by Christina Curtis. <em>Bioinformatics</em> 34 (2): 330–37. <a href="https://doi.org/10.1093/bioinformatics/btx604">https://doi.org/10.1093/bioinformatics/btx604</a>.
</div>
<div id="ref-kimExtrachromosomalDNAAssociated2020" class="csl-entry">
Kim, Hoon, Nam-Phuong Nguyen, Kristen Turner, Sihan Wu, Amit D. Gujar, Jens Luebeck, Jihe Liu, et al. 2020. <span>“Extrachromosomal <span>DNA</span> Is Associated with Oncogene Amplification and Poor Outcome Across Multiple Cancers.”</span> <em>Nature Genetics</em> 52 (9): 891–97. <a href="https://doi.org/10.1038/s41588-020-0678-2">https://doi.org/10.1038/s41588-020-0678-2</a>.
</div>
<div id="ref-martinez-jimenezCompendiumMutationalCancer2020" class="csl-entry">
Martínez-Jiménez, Francisco, Ferran Muiños, Inés Sentís, Jordi Deu-Pons, Iker Reyes-Salazar, Claudia Arnedo-Pac, Loris Mularoni, et al. 2020. <span>“A Compendium of Mutational Cancer Driver Genes.”</span> <em>Nature Reviews Cancer</em> 20 (10): 555–72. <a href="https://doi.org/10.1038/s41568-020-0290-x">https://doi.org/10.1038/s41568-020-0290-x</a>.
</div>
<div id="ref-mauraPracticalGuideMutational2019" class="csl-entry">
Maura, Francesco, Andrea Degasperi, Ferran Nadeu, Daniel Leongamornlert, Helen Davies, Luiza Moore, Romina Royo, et al. 2019. <span>“A Practical Guide for Mutational Signature Analysis in Hematological Malignancies.”</span> <em>Nature Communications</em> 10 (1): 2969. <a href="https://doi.org/10.1038/s41467-019-11037-8">https://doi.org/10.1038/s41467-019-11037-8</a>.
</div>
<div id="ref-nguyenPancancerLandscapeHomologous2020" class="csl-entry">
Nguyen, Luan, John Martens, Arne Van Hoeck, and Edwin Cuppen. 2020. <span>“Pan-Cancer Landscape of Homologous Recombination Deficiency.”</span> Preprint. <span>Cancer Biology</span>. <a href="https://doi.org/10.1101/2020.01.13.905026">https://doi.org/10.1101/2020.01.13.905026</a>.
</div>
<div id="ref-pcawgmutationalsignaturesworkinggroupRepertoireMutationalSignatures2020" class="csl-entry">
PCAWG Mutational Signatures Working Group, PCAWG Consortium, Ludmil B. Alexandrov, Jaegil Kim, Nicholas J. Haradhvala, Mi Ni Huang, Alvin Wei Tian Ng, et al. 2020. <span>“The Repertoire of Mutational Signatures in Human Cancer.”</span> <em>Nature</em> 578 (7793): 94–101. <a href="https://doi.org/10.1038/s41586-020-1943-3">https://doi.org/10.1038/s41586-020-1943-3</a>.
</div>
<div id="ref-pcawg-structuralvariationworkinggroupGenomicFootprintsActivated2020" class="csl-entry">
PCAWG-Structural Variation Working Group, PCAWG Consortium, Lina Sieverling, Chen Hong, Sandra D. Koser, Philip Ginsbach, Kortine Kleinheinz, et al. 2020. <span>“Genomic Footprints of Activated Telomere Maintenance Mechanisms in Cancer.”</span> <em>Nature Communications</em> 11 (1): 733. <a href="https://doi.org/10.1038/s41467-019-13824-9">https://doi.org/10.1038/s41467-019-13824-9</a>.
</div>
</div>
</div>
</div>

   
   
              </div>
  </div>
  </div>
  </div>
   
      

  <script>
    $(document).ready(function () {

		// add bootstrap table styles to pandoc tables
	$('tr.header').parent('thead').parent('table').addClass('table table-condensed');
		
 		
	    });
  </script>



    <!-- dynamically load mathjax for compatibility with self-contained -->
  <script>
    (function () {
	var script = document.createElement("script");
	script.type = "text/javascript";
	script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
	document.getElementsByTagName("head")[0].appendChild(script);
    })();
  </script>
  
</body>
</html>
